<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Population Genomics</title>
    <meta charset="utf-8" />
    <meta name="author" content="Jinliang Yang" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="../../asset/unl-150.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Population Genomics
### Jinliang Yang
### Oct. 212022

---




# About Me

- Bachelor (2005) and Master (2008) degree from CAU!

- About 15 years experience in maize genetics

--

### Prefessional Experience

- 2008-2014: __Ph.D.__ @ Iowa State University 
  - Genetics and Genomics
  
- 2014-2017: __PostDoc__ @ University of California, Davis 
  - Evolutionary Genetics

- 2017-Now: __Assistant Professor__ @ University of Nebraska-Lincoln
  - Quantitative Genetics and Statistical Genomics

---

# About my lab

- My lab focuses on __quantitative genetics__ and __statistical genomics__ of maize and its wild relatives

  - From historical _domestication_ to future _improvement_.

--

&lt;div align="center"&gt;
&lt;img src="study.png" height=200&gt;
&lt;img src="qrcode1.png" height=200&gt;
&lt;/div&gt; 

--

  - Lab website: https://jyanglab.com/
    - TEACHING: Courses
      - Click the [Module]()
      
  - Links for the slides: 

---

# Goals and Objectives

### Overall goal: 

Use maize as a model species to illustrate basic principles of crop domestication and its implication for crop improvement

--

### Objectives:

- 1) Understand basic population genomics terminology

- 2) Learn how to measure diversity using sequencing data

- 3) Understand the statistics for measuring population differentiation

- 4) Be able to scan for direct and linked selection

---

# A timely topic

### __Nobel Prize in Physiology or Medicine 2022__

.pull-left[
&lt;div align="center"&gt;
&lt;img src="paabo.png" height=350&gt;
&lt;/div&gt;
]

--

.pull-right[
&lt;div align="center"&gt;
&lt;img src="ancientcorn.png" height=100&gt;
&lt;/div&gt;
A maize cob from the Ocampo Caves (Valenzuela cave), dated to 3890 ± 60 years before the present. Length, 47 mm. 
&gt; from Paabo's 2003 Science paper
]

--

-----------

__Svante Paabo__ represents the field of _evolutionary genetics_ and/or _population genetics!!!_

---
# A job for Google X


---

# Syllabus

### Module 1: Introduction and popgen terminology
- Introduction of population genomics
- 2. something

### Module 2: Diversity measurement
- topic 1
- topic 2

### Module 3: Scan for direct and linked selection
- topic 1
- topic 2


---
# 151 crops and their domestication origins

&lt;div align="center"&gt;
&lt;img src="crops.png" height=450&gt;
&lt;/div&gt;

---
# Maize domestication


&lt;div align="center"&gt;
&lt;img src="ca.png" height=150&gt;
&lt;/div&gt;


--

---------

.pull-left[
&lt;div align="center"&gt;
&lt;img src="shelter.png" height=200&gt;
&lt;/div&gt;
]

.pull-right[
&lt;div align="center"&gt;
&lt;img src="stone.png" height=200&gt;
&lt;/div&gt;
]

This __8,700__ year-old milling stone (excavated in an area of __the Balsas Valley__ in southwestern Mexico) was used to process maize and other crops. Maize starch grains were recovered from the stone.


---
# Plant Domestication

&lt;div align="center"&gt;
&lt;img src="Teosinte.png" height=200&gt;
&lt;/div&gt;

- “Plant domestication is the _genetic modification_ of a wild species to create a new form of a plant altered _to meet human needs_.” 
&gt; Doebley et al., 2006

--

__Plant domestication__ is the process whereby wild plants have been evolved into crop plants through artificial selection. 

- This usually involves an early hybridization event followed by selective breeding.

---
# Wild and domesticated species

&lt;div align="center"&gt;
&lt;img src="wild.png" height=300&gt;
&lt;/div&gt;

--

Fitness are different between wild ancestors and domesticated crops

- Fitness describes individual __reproductive__ success.

---

# Domestication Syndrome

.pull-left[
&lt;div align="center"&gt;
&lt;img src="dom.png" height=500&gt;
&lt;/div&gt;
]

.pull-left[
Compared to their progenitors, crops typically have:
- Larger grains or fruits

- More robust plants overall

- More determinate growth or increased apical dominance  

- A loss of natural seed dispersal

&gt; Doebley et al., 2006
]

---
# Genes controlling domestication traits

&lt;div align="center"&gt;
&lt;img src="dom_genes.png" height=300&gt;
&lt;/div&gt;
&gt; Doebley et al., 2006



---
# Population genomics for domestication

### Understand the popgen
- To identify the selective sweeps 
  - [Hufford et al., 2012](https://www.nature.com/articles/ng.2309) Nature Genetics
- To infer the history of populations
  - [Beissinger et al., 2016](https://www.nature.com/articles/nplants201684) Nature Plants
- To examine the relative effects of evolutionary forces across the genome 
  - [Xu et al., 2020](https://www.nature.com/articles/s41467-020-19333-4) Nature Communications
  
--

### Design a selection scan experiment
- Sampling wild and domesticated populations
- DNA sequencing and data analysis


---
# Positive selection

Refers to a history of a locus in which advantageous mutations have arisen and fixed or are in the process of fixing.
- Advantageous mutations much rarer than deleterious mutations.
- Advantageous mutation can rapidly increase in frequency.
- Patterns generated by this rapid rise are used to detect signatures of positive selection.

---
# Negative (purifying) selection

Refers to a history of a locus in which the vast majority of mutations that have arisen have been deleterious.
- A gene or noncoding region under negative selection is often conserved.
- These genes or regions are constrained by selection.
- Relatively common

---
# Balancing selection

- Alleles under balancing selection are not universally advantageous or deleterious.
- Balanced polymorphisms consist of alleles whose fitness changes with time, space, or population frequency. 
  - Heterozygote advantage (or overdominant) selection: heterozygotes genotypes have higher fitness than either homozygous genotype
  - Spatially or temporally varying selection: Fitness of alleles in a population is dependent on the environment or season



  
---

.pull-left[
&lt;div align="center"&gt;
&lt;img src="fisher.jpg" height=150&gt;
&lt;/div&gt;
- Lab website: https://jyanglab.com/
]

.pull-right[
&lt;div align="center"&gt;
&lt;img src="Karl_Pearson_1912.jpg" height=150&gt;
&lt;/div&gt;
- The early geneticist __Karl Pearson__ originated the concepts of regression and correlation.
]












---
# binomial distribution

The binomial distribution is a discrete probability distribution. It describes the outcome of n independent trials in an experiment. Each trial is assumed to have only two outcomes, either success or failure. If the probability of a successful trial is p, then the probability of having x successful outcomes in an experiment of n independent trials is as follows.

`\begin{align*}
\mathbf{y} &amp;= \mathbf{X}\mathbf{b} + \boldsymbol{\epsilon} \\
\end{align*}`


---
# Wright Fisher simulation

Consider a single nucleotide position with two alleles, `\(A_1\)` and `\(A_2\)`

In generation `\(t\)` there are `\(x\)` individuals carrying allele `\(A_1\)`, which is at frequency `\(p_t = x/2N\)`.

This implies that there are `\(2N - x\)` individuals carrying allele `\(A_2\)`, which is at frequency `\(q_t = 1- p_t= 1- x/2N\)`..

The sampling of alleles for the next generations is equivalent to sampling from a binomial distribution with parameters size = `\(2N\)` and prob= `\(x/2N\)`.

Therefore, the mean and variance of `\(p\)` in the next generation for the Wright-Fisher model are:


`\begin{align*}
E(p_t) &amp;= p_t \\
Var(p_{t+1}) &amp;= p_tq_t/2N 
\end{align*}`


---

# Wright Fisher simulation



```r
wright_fisher &lt;- function(N=1000, t=100, A1=100){
  # N: number of diploid individuals. [N=1000, integer]
  # t: number of generations. [t=100, numeric]
  # A1: number of A1 (the minor allele) in the first generation. [A1=10, integer]

  # create a vector 
  x[1] = A1

  # Start to loop over m generations
  # binomial sampling at each generation - determines allele count in next generations
  for (i in 2:t){
    # calculate allele freq in the current generation
    k &lt;- (x[i-1])/(2*N)
    # generate a vector with 2N individuals
    n=seq(0,2*N,1)
    
    # density for the binomial distribution with paramters size=2N and prob=k
    prob=dbinom(n, 2*N, k)
    # plot(prob)
    # choose one from a vector each with certain prob
    x[i]=sample(0:(2*N), 1, prob=prob)
  }
  # return back the results
  return(x)
}
```

---


```r
set.seed(12345)
sim1 &lt;- wright_fisher(N=50, t=100, A1=20)
plot(sim1[1:100], type="o", pch=19, xlab="Generations", ylab="A1 allele count")

sim1 &lt;- wright_fisher(N=50, t=100, A1=20)
plot(sim1[1:100], type="o", pch=19, xlab="Generations", ylab="A1 allele count")
```

Linear models form the backbone of most estimation procedures in plant breeding and quantitative genetics.

## Multiple regression

Most common type of linear model
- Predictor variables represent observed values for a set of continuous variables.

## Analysis of variance
Model containing only indicator variables.
- Categories
- Wide array of classification problems can be handled.

---
# A half-sib design













    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
