<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Heritability and repeatability</title>
    <meta charset="utf-8" />
    <meta name="author" content="Jinliang Yang" />
    <meta name="date" content="2022-10-24" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="../../asset/unl-150.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Heritability and repeatability
### Jinliang Yang
### Oct.Â 24, 2022

---





# Additive and dominance variance


| Genotype  | Freq      | Breeding Value | `\(A^2\)`  | Dominance Deviation  | `\(D^2\)` |
| :-------: | :-------: | :-----------: | :-------: | :-------: | :-------: | :-------: |
| `\(A_1A_1\)`  | `\(p^2\)`     | `\(2q\alpha\)`    | `\((2q\alpha)^2\)`  |  `\(-2q^2d\)`   | `\((-2q^2d)^2\)` |
| `\(A_1A_2\)`  | `\(2pq\)`     | `\((q-p)\alpha\)` | `\((q-p)^2\alpha^2\)` |  `\(2pqd\)`   | `\((2pqd)^2\)` |
| `\(A_2A_2\)`  | `\(q^2\)`     | `\(-2p\alpha\)`   | `\((-2p\alpha)^2\)` |  `\(-2p^2d\)`   | `\((-2p^2d)^2\)` |



The additive and dominance genetic variance in a HWE population is:

`\begin{align*}
\sigma_A^2 &amp; = 2pq(a + d(q-p))^2 \\
\sigma_D^2 &amp; = (2pqd)^2 \\
\end{align*}`
  
--

Because `\(G = A + D\)` for single locus, 

`\begin{align*}
\sigma_G^2 &amp; = \sigma_A^2 + \sigma_D^2 +  2\sigma_{A, D}
\end{align*}`

Then `\(\sigma_{A, D}=?\)`

---
# Additive and dominance variance


| Genotype  | Freq      | Breeding Value | `\(A^2\)`  | Dominance Deviation  | `\(D^2\)` |
| :-------: | :-------: | :-----------: | :-------: | :-------: | :-------: | :-------: |
| `\(A_1A_1\)`  | `\(p^2\)`     | `\(2q\alpha\)`    | `\((2q\alpha)^2\)`  |  `\(-2q^2d\)`   | `\((-2q^2d)^2\)` |
| `\(A_1A_2\)`  | `\(2pq\)`     | `\((q-p)\alpha\)` | `\((q-p)^2\alpha^2\)` |  `\(2pqd\)`   | `\((2pqd)^2\)` |
| `\(A_2A_2\)`  | `\(q^2\)`     | `\(-2p\alpha\)`   | `\((-2p\alpha)^2\)` |  `\(-2p^2d\)`   | `\((-2p^2d)^2\)` |

### `\(\sigma_{A, D}=?\)`

$$
`\begin{aligned}
Cov(X, Y) &amp; = \sigma_{XY} \\
&amp; = E([X - E(X)][Y - E(Y)]) \\
&amp; = E(XY) - E(X)E(Y) \\
\end{aligned}`
$$

where,

$$
`\begin{aligned}
E(XY) = \sum_i \sum_j x_i y_j Pr(X = x_i, Y = y_j)
\end{aligned}`
$$

---
# Additive and dominance variance


| Genotype  | Freq      | Breeding Value | `\(A^2\)`  | Dominance Deviation  | `\(D^2\)` |
| :-------: | :-------: | :-----------: | :-------: | :-------: | :-------: | :-------: |
| `\(A_1A_1\)`  | `\(p^2\)`     | `\(2q\alpha\)`    | `\((2q\alpha)^2\)`  |  `\(-2q^2d\)`   | `\((-2q^2d)^2\)` |
| `\(A_1A_2\)`  | `\(2pq\)`     | `\((q-p)\alpha\)` | `\((q-p)^2\alpha^2\)` |  `\(2pqd\)`   | `\((2pqd)^2\)` |
| `\(A_2A_2\)`  | `\(q^2\)`     | `\(-2p\alpha\)`   | `\((-2p\alpha)^2\)` |  `\(-2p^2d\)`   | `\((-2p^2d)^2\)` |

### `\(\sigma_{A, D}=0\)`  


`\begin{align*}
E(XY) &amp; = \sum_i \sum_j x_i y_j Pr(X = x_i, Y = y_j) \\
E(AD) &amp;= p^2\times (2q\alpha)\times (-2q^2d) + 2pq \times (q-p)\alpha \times 2pqd + q^2 \times (-2p\alpha) \times (-2p^2d) \\
      &amp;= 4\alpha d p^2q^2(-q + q -p +p) \\
      &amp;=0
\end{align*}`

---

# Variance due to GxE interaction

In this case, 

`\begin{align*}
&amp; P = G + E + I_{GE} \\
\end{align*}`

where `\(I_{GE}\)` is the interaction effect between genotype and environment.

--

And the variance is,

`\begin{align*}
Var(P) &amp; =  Var(G + E + I_{GE}) \\
       &amp; =  Var((G + E) + I_{GE}) \\
\end{align*}`

--

`\begin{align*}
\sigma_P^2 &amp; = \sigma^2_G + \sigma^2_E + 2Cov(G, E) + \sigma^2_{I_{GE}} \\
\end{align*}`

- The covariance between genetic and non-genetic effects is often zero except in special circumstances.

---
# Variance due to GxE interaction

`\begin{align*}
\sigma_P^2 &amp; = \sigma^2_G + \sigma^2_E + 2Cov(G, E) + \sigma^2_{I_{GE}} \\
\end{align*}`

#### `\(Cov(G, E)\)` or `\(\sigma_{G,E}\)`
- Refers to **genotype-environment covariance**
- A measure of the physical association of particular genotypes with particular environmental general effects
- If individuals are __randomly distributed__ with respect to macroenvironments, then `\(\sigma_{G,E}=0\)`

--

#### `\(\sigma^2_{I_{GE}}\)`
- Refers to **variation of genotype-environment interaction**
- If G and E are nonadditive, it measures how large the variation is.

---

## Example: salt marsh cord grass

- An experiment was conducted using a __clonal plant__ (Silander,1985) in three sites --- dune, swale, marsh.

&lt;table&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;"&gt; Trait &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; Var_G &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; Va_E &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; Var_I &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; Var_e &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; Tillers per plant &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 34 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 20 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 45 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; Plant hight &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 13 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 57 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 7 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 23 &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

The observed components of variance in above table as percentages of the total phenotypic variance.

--

- `\(Var(G)\)`: the among-clone variance 
  - is an estimate of the total genetic variance
- `\(Var(E)\)`: the variance among the three sites 
  - is an estimate of the general environmental variance
- `\(Var(I)\)`: the clone `\(\times\)` site variance
- `\(Var(e)\)`: the variance within sites 
  - more specifically, among members of the same clones within sites

--

Here, because individual plants were __distributed randomly__ within sites, `\(\sigma_{G,E}=0\)`.



---
# Sources of variation

Several sources of variation that reduce the precision of the experiment:

--
- Environmental factors (among sites)
  - climatic, nutritional, etc

--

- Measurement error (within site, among clones)
  - microenvironment
  - unrecognized causes (developmental accidents)

--

### Two types of non-genetic variance

1. **General environmental variance `\(\sigma_{Eg}^2\)`**: non-genetic variance contributing to the between-individual variation that is permanent. (Maternal effect, barn effect)

2. **Special environmental variance `\(\sigma_{Es}^2\)`**: within individual variation arising from temporary or localized circumstances. (Measurement error, developmental accidents, temporary climactic effects, etc.)



---

# Repeatability of measurements

The __repeatability__ is the proportion of total variance of single measurements that is due to __permanent differences between individuals__.

`\begin{align*}
&amp; r = \frac{\sigma_G^2 + \sigma_{Eg}^2}{\sigma_P^2} \\
\end{align*}`

--


### Broad sense heritability ( `\(H^2\)` )

`\begin{align*}
&amp; H^2 = \frac{\sigma_G^2}{\sigma_P^2} \\
\end{align*}`

--

Repeatability sets an upper limit to **heritability**.

---

# Introduction to heritability

### Broad sense heritability ( `\(H^2\)` )

`\begin{align*}
&amp; H^2 = \frac{\sigma_G^2}{\sigma_P^2} \\
\end{align*}`



- Proportion of variation due to genotypic effects.

- This represents __nature__ versus __nurture__.

--

- In a breeding program, this measure of heritability is useful when all forms of genetic variation can be exploited
  - such as selecting among a set of clones to propagate a cultivar.

---

# Introduction to heritability

### Narrow sense heritability ( `\(h^2\)` )

`\begin{align*}
&amp; h^2 = \frac{\sigma_A^2}{\sigma_P^2} \\
\end{align*}`

- Proportion of phenotypic variation due to variation in __breeding values__.

--

- Since parents pass on alleles, not genotypes, `\(h^2\)` is more meaningful in determining amount of genetic progress from generation to generation due to selection and intermating



---

# Repeatability of measurements

One way to increase the accuracy of measurements is to use the average of multiple measurements instead of just one.

According to F&amp;M, formula [8.14]
`\begin{align*}
&amp; \sigma_{P(n)}^2 = \sigma_G^2 + \sigma_{Eg}^2 + \frac{1}{n}\sigma_{Es}^2 \\
\end{align*}`

The contribution of `\(\sigma_{Es}^2\)` to the total variance is reduced proportionally by the number of measurements taken.

This means that averages of multiple measurements are more accurate indicators of the underlying genetic value than single measurements.

---

# Repeatability of measurements

The repeatability of the average of multiple measurements is:

`\begin{align*}
r_n &amp; = \frac{\sigma_G^2 + \sigma_{Eg}^2}{\sigma_{P(n)}^2} \\
&amp; = 1- \frac{\frac{1}{n}\sigma_{Es}^2}{\sigma_{P(n)}^2} \\
\end{align*}`

If repeatability is low, taking multiple measurements and avaraging them can really improve accuracy.


--

However,

- Repeated measurement may not suitable for behavioral and physiological traits because of the temporal organismal changes.




---

# Summary of variance partitioning

### Phenotypic variance

- Genetic componenents
  1. Additive
  2. Non-additive

- Environmental componenents
  1. General
  2. Special
  
--

| Data                            | Parition      | Ratio | Definition |
| :-----------------------------  | :-------: | :-----------: | :---: |
| Relatives   | `\({V_A}:({V_{NA} + V_{Eg} + V_{Es}})\)`     | `\(\frac{V_A}{V_{P}}\)`     | `\(h^2\)` |
| Genetically uniform group       | `\(({V_A} + V_{NA}):(V_{Eg} + V_{Es})\)`     | `\(\frac{V_G}{V_{P}}\)` |  `\(H^2\)` |
| Multiple measurements           | `\((V_G + V_{Eg}):V_{Es}\)`     | `\(\frac{V_G + V_{Eg}}{V_{P}}\)`   |   `\(r\)` |
















    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
