<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Population Genomics Module 3</title>
    <meta charset="utf-8" />
    <meta name="author" content="Jinliang Yang" />
    <meta name="date" content="2022-11-03" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="../../asset/unl-150.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Population Genomics Module 3
### Jinliang Yang
### Nov. 3, 2022

---





# Syllabus

### Module 1: Introduction and popgen terminology
- Introduction of population genomics
- Basic principles of evolutionary processes

### Module 2: Diversity measurement
- Heterozygosity and diversity
- Population differentiation ( `\(F_{ST}\)` )

--

### __Module 3: Scan for direct and linked selection__
- __Direct selection__
- __Linked selection__

---

# Slides for today's class module

- Scan QR code to view the HTML slides: 

&lt;div align="center"&gt;
&lt;img src="qr-code.png" height=300&gt;
&lt;/div&gt; 

--

- Lab website: https://jyanglab.com/
  - TEACHING: Courses
    - __Module 3__: [[HTML](https://jyanglab.com/slides/2022-module/week3/w3.html)]
      
---

# Neutral theory of molecular evolution

"The neutral theory asserts that the great majority of evolutionary changes at the molecular level are caused
NOT by _Darwinian selection_ 
but by _random drift of selectively neutral or nearly neutral mutants_."

&gt; Motoo Kimura (木村 資生), 1983 
&gt; - Iowa State with Jay Lush and then University of Wisconsin with James Crow


--

### Core ideas of neutral theory of molecular evolution:

- #### Most mutations are not advantageous

  - Selectively (or effectively) neutral if `\(s &lt; 1/2N_e\)`
  
- #### Most changes that are fixed over time are selectively neutral (fixed by drift)

   - Drift rather than selection predominates
---

# Neutral Theory

### What the neutral theory does not claim

- __Does NOT claim__ natural selection is unimportant in evolution

  - In fact, most morphological adaptations are the result of natural selection 

--

- It __does NOT deny__ that most mutations are (slightly) deleterious (it claims most of the variation _we see_ is neutral)

  - Most of the deleterious mutations have been eliminated
  
  - Rare mutations have been fixed


--

### Selection counteracts drift

- `\(s &gt; 1/2N_e\)`

`\begin{align*}
Pr(fix) = \frac{1 - e^{-2s}}{1-e^{-4N_es}}
\end{align*}`


---




```r
set.seed(12347)
Ne=20; A1=1; t=4*Ne
frq &lt;- wright_fisher(N=Ne, A1=A1, t=t)
plot(frq, type="l", ylim=c(0,1), col=3, xlab="Generations", ylab="Freq")
    for(u in 1:100){
      frq &lt;- wright_fisher(N=N, A1=A1, t=t)
      random &lt;- sample(1:1000,1,replace=F)
      randomcolor &lt;- colors()[random] 
      lines(frq, type="l", lwd=3, col=(randomcolor))
    }
```

&lt;img src="w3_files/figure-html/unnamed-chunk-2-1.png" style="display: block; margin: auto;" /&gt;

---

# Expected allele frequencies distribution

On timescales shorter than those required for mutations to fix, selection will change the mean frequency of alleles in a population.

--

For new mutations, the density of polymorphisms found at frequency `\(q\)`, is


`\begin{align*}
f(q) &amp; = \frac{2 \mu}{q(1-q)} \frac{1 - e^{(-4N_es)(1-q)}}{1 - e^{(-4N_e s)}} \\
\end{align*}`

&gt; Wright, 1969

- Where `\(\mu\)` is the mutation rate.
- `\(s\)` is the fitness effect. 
  - Advantageous mutations have `\(s &gt; 0\)` and deleterious mutations have `\(s &lt;0\)`



---

# The expected frequency spectra

`\begin{align*}
f(q) &amp; = \frac{2 \mu}{q(1-q)} \frac{1 - e^{(-4N_es)(1-q)}}{1 - e^{(-4N_e s)}} \\
\end{align*}`



```r
# expected freq spectra
f &lt;- function(q, ns){
  frq = 2/(q*(1-q)) * (1 - exp(-4*ns*(1-q))) / (1 - exp(-4*ns))
  return(frq)}
q &lt;- seq(from = 0.01, to =0.99, by=0.01)

## Ploting function
plot(q, f(q, ns=0.01), type="l", lty=1, lwd=3, xlab="Ns", ylab="No. of polymorhpic sites", cex.lab=2)
lines(q, f(q, ns=-50), type="l", lty=1, lwd=3, col="red")
lines(q, f(q, ns=-5), type="l", lty=2, lwd=3, col="red")
lines(q, f(q, ns=5), type="l", lty=1, lwd=3, col="blue")
lines(q, f(q, ns=50), type="l", lty=2, lwd=3, col="blue")
legend(0.6, 200, title="Ne*s", legend=c("-50", "5", "0", "-5", "50"), 
       col=c("red", "red", "black", "blue", "blue"), 
       lty=c(1,2,1,1,2), cex=2, lwd=3)
```

---

# The expected distribution of `\(f(q)\)`

`\begin{align*}
f(q) &amp; = \frac{2 \mu}{q(1-q)} \frac{1 - e^{(-4N_es)(1-q)}}{1 - e^{(-4N_e s)}} \\
\end{align*}`

.pull-left[
&lt;img src="w3_files/figure-html/unnamed-chunk-4-1.png" width="100%" style="display: block; margin: auto;" /&gt;
]


--

.pull-right[
- #### Deleterious alleles =&gt; lower frequencies
 - most strongly deleterious mutations are immediately removed from the population
 
- #### Advantage alleles shifted toward higher frequencies
  - most strongly advantageous mutations fix very rapidly.
]



---

# Types of selection

To find loci that are under selection we test for departures from the neutral theory

--

### Purifying selection: 
  - Deleterious mutations are eliminated

### Positive selection: 
  - Opposite of purifying
  - Favorable mutations are selected

### Balancing selection: 
  - Maintains two or more variants at a locus


---

# Genetic diversity within pops

### Expected diversity 
- Number of alleles/locus (allelic richness) 
- Polymorphism (loci with &gt; 1 allele)
- Theta ( `\(\theta\)` ) = `\(4N_e \mu\)`
  - `\(N_e\)` = effective population size
  - `\(\mu\)` = mutation rate per generation

--

### Expected heterozygosity

`\(H_{exp}\)` = 1 - (avg expected __homozygosity__ over all loci)

`\begin{align*}
H_{exp} = 1 - \frac{1}{m}\sum_{k=l}^{m} \sum_{i=l}^{k} p_{ki}^2
\end{align*}`
  - `\(m\)` is the number of loci
  - `\(k\)` is the number of alleles at a particular locus
  - `\(p_{ki}\)` is the frq of `\(i^{th}\)` allele at `\(k^{th}\)` locus

# Common estimators of `\(\theta\)` under the infinite sites model

For an individual SNP, one allele has sample frequency `\(p\)`, alternative allele frequency is `\(q\)`, such that `\(p +q =1\)`.

--

__Heterozygosity__ at this SNP site is:

`\begin{align*}
h = \frac{n}{n-1}(1 - p^2 - q^2)
\end{align*}`

- where `\(n\)` is the number of sequences in the sample. 

---

# Under the infinite sites model

`\begin{align*}
\pi = &amp; \sum_{j=1}^{S}h_j \\
\end{align*}`

- Where `\(S\)` is the number of segregating sites
- `\(h_j\)` is the heterozygosity at the `\(j\)`th SNP site.

--

Under the __infinite sites model__ for a diploid population at HWE,

`\begin{align*}
E(\pi) = &amp; \theta = 4N_e \mu\\
\end{align*}`

Which is why this statistic is sometimes called `\(\theta_\pi\)`.


---
# An alternative method: Watterson' theta

In this method, we summarize SNPs using the total number of segregating sites, `\(S\)`, in the sample.

However, because larger sample sizes will result in larger values of `\(S\)`, we must adjust the statistic to be

`\begin{align*}
\theta_W = &amp; \frac{S}{a} \\
\end{align*}`

Where `\(a\)` is,

`\begin{align*}
a=\sum_{i=1}^{n-1}\frac{1}{i}
\end{align*}`
  - `\(n\)` is the number of samples

Or, combine them together

`\begin{align*}
\theta_W = &amp; \frac{S}{\sum_{i=1}^{n-1}\frac{1}{i}} \\
\end{align*}`

---

# An Alternative method: Watterson' theta

&lt;div align="center"&gt;
&lt;img src="p4.png" height=150&gt;
&lt;/div&gt;


`\begin{align*}
\theta_W = &amp; \frac{S}{\sum_{i=1}^{n-1}\frac{1}{i}} \\
= &amp; 2/(1/1 + 1/2 + 1/3) \\
= &amp; 1.09
\end{align*}`

The per site measure would be `\(1.09/10=0.109\)`, which is very similar to `\(\theta_\pi\)`


---
# The frequency spectrum of alleles

&lt;div align="center"&gt;
&lt;img src="p10.png" height=200&gt;
&lt;/div&gt;


In this case, among these 10 haplotypes are 10 segregating sites, each of which can have a frequency between `\(1/n\)` and `\((n-1)/n\)`

--

Visually summarize the MAF of all segregating sites using the __allele frequency spectrum__
  

---
# The frequency spectrum of alleles

The allele frequency specturm is also referred to as the __site frequency spectrum (SFS)__.

.pull-left[
&lt;div align="center"&gt;
&lt;img src="p10.png" height=300&gt;
&lt;/div&gt;
]

--

.pull-left[

```r
maf &lt;- c(0.1, 0.1, 0.3, 0.1, 0.3, 
         0.2, 0.1, 0.4, 0.1, 0.1)
sfs &lt;- table(maf)
barplot(sfs, col="#cdc0b0", xlab="Minor allele frequency", 
        ylab="No. of segregating sites", 
        cex.axis =1.5, cex.names = 1.5, cex.lab=1.5)
```

&lt;img src="w3_files/figure-html/unnamed-chunk-5-1.png" width="80%" style="display: block; margin: auto;" /&gt;
]

---
# The frequency spectrum of alleles

- The allele frequency specturm is also referred to as the __site frequency spectrum (SFS)__.
- Use the sequences of one or more closely related species, we can get the ancestral state.
- Therefore, we can describe variation at each site using the __derived allele frequency (DAF)__.

.pull-left[
&lt;div align="center"&gt;
&lt;img src="daf.png" height=300&gt;
&lt;/div&gt;
]

--

.pull-left[

```r
maf &lt;- c(0.1, 0.1, 0.3, 0.1, 0.3, 
         0.2, 0.1, 0.6, 0.9, 0.9)
sfs &lt;- table(maf)
barplot(sfs, col="#cdc0b0", xlab="Derived allele frequency", ylab="No. of segregating sites", 
        cex.axis =1.5, cex.names = 1.5, cex.lab=1.5)
```

&lt;img src="w3_files/figure-html/unnamed-chunk-6-1.png" width="50%" style="display: block; margin: auto;" /&gt;
]

---

# Signature of negative selection


.pull-left[
&lt;div align="center"&gt;
&lt;img src="negsel.png" height=300&gt;
&lt;/div&gt;
]

--

.pull-left[
- Comparison of expected and observed is __uneven__

- The rare alleles are at lower freq than expected

- Evidence of __negative selection__ (or __purifying selection__)

- However, confounded by population demographics (i.e., bottleneck effect)
]

---

# Signature of positive/balancing selection


.pull-left[
&lt;div align="center"&gt;
&lt;img src="possel.png" height=300&gt;
&lt;/div&gt;
]

--

.pull-left[
- Comparison of expected and observed is __too even__

- The most common allele is more common than expected

- Evidence of __positive selection__ or __balancing selection__

- However, confounded by population demographics (i.e., population expansion)
]


---

# Diversity measurement

We now consider several statistics summarizing sequencing diversity that use information about __the frequency of derived alleles__
  - As these capture more information about our sequencing data.

--

Fu and Li (1993) defined a statistic, `\(\epsilon_1\)`, based on the number of __derived singletons__ in a sample.

`\begin{align*}
\epsilon_1 = S_1 \\
\end{align*}`

- Where `\(S_1\)` is the number of segregating site with derived alleles found on only one haplotype.

--

If we don't know the ancestral status, we can aslo define a statistic, `\(\eta_1\)`, based on __all singletons__ in a sample

`\begin{align*}
\eta_1 = S_1^*\frac{n-1}{n} \\
\end{align*}`

- Where `\(S_1^*\)` is all the singletons.

---
# Diversity measurement

A second summary statistic of diversity that uses ancestral state information is `\(\theta_H\)`:

`\begin{align*}
\theta_H = \frac{\sum_{i=1}^{n-1} i^2S_i}{n(n-1)/2} \\
\end{align*}`

- Where `\(S_i\)` is again the number of segregating sites where `\(i\)` haplotypes carry the derived allele (Fay and Wu, 2000).

--

.pull-left[
&lt;div align="center"&gt;
&lt;img src="daf.png" height=300&gt;
&lt;/div&gt;
]

--

.pull-left[
`\begin{align*}
\theta_H = &amp; \frac{\sum_{i=1}^{n-1} i^2S_i}{n(n-1)/2} \\
= &amp; \frac{(1^2 \times 4 + 2^2 \times 1 + 3^2 \times 2 + 6^2 \times 1 + 9^2 \times 2)}{10(10-1)/2} \\
= &amp; 4.98
\end{align*}`
]

---

# Summary of the `\(\theta\)` statistics

All of these statistics --- `\(\epsilon_1, \eta_1, \theta_H\)` --- are estimators of `\(\theta\)` 
- at __mutation-drift__ equilibrium 
- under an __infinite sites__ mutational model


--

Specifically,

`\begin{align*}
E(\epsilon_1) = E(\eta_1) = E(\theta_H)
\end{align*}`

These relationships arise because we know the expected shape of the allele frequency distribution under our standard neutral assumptions.



---

# Detecting selection using the SFS

## The effects of positive selection


.pull-left[
&lt;div align="center"&gt;
&lt;img src="sfs1.png" height=300&gt;
&lt;/div&gt;
&gt; Hanh, 2020
]

.pull-right[
- After sweep ended, new mutations started to accumulate.

- These new mutations are by definition __singletons__
 - there is only one origin in the sample with each derived allele.
]

The SFS can be skewed toward an excess of low-frequency polymorphisms relateive to the neutral spectrum.

---
# Detecting selection using the SFS

## The effects of balancing selection

Here we consider a simple scenario with a single biallelic site that has been under balancing selection for a long time.
- Variation within each allelic class has been able to __build up__ and __reach equilibrium__

.pull-left[
&lt;div align="center"&gt;
&lt;img src="sfs3.png" height=200&gt;
&lt;/div&gt;
&gt; Bitarello et al., 2018
]

.pull-right[
- Neutral mutations has accumulated both within and between allelic classes

- Overall variation is higher

- SNPs at intermediate frequency show __a distinctive "bump"__ in the SFS.
]


---
# Detecting selection using SFS

A straightforward way would be test a difference between two SFSs.
- However, linkage among sites means that __SNPs at a locus are not independent__, which violates the assumptions made by almost all such test.

--

### Instead, we use `\(\theta\)` to detect deviations.

- `\(\theta_\pi\)`: pairwise necleotide diversity.

--

- `\(\theta_W\)`: Watterson's `\(\theta\)`, using total number of segregating sites

--

- `\(\epsilon_1 = S_1\)`: the number of derived singletons in a sample.
- `\(\eta_1\)`: based on all singletons in a sample.

--

Under the standard neutral model, all of these test statistics are expected to have a mean of 0.

---

# Tajima's D and related tests

Tajima (1989) constructed the first test to detect difference between the SFS. 

His statistic, `\(D\)`, was defined as:

`\begin{align*}
D = \frac{\theta_\pi - \theta_W}{\sqrt{Var(\theta_\pi - \theta_W)}}
\end{align*}`

--

Fu and Li (1993) created similar statistics. These are known as Fu and Li's `\(D\)`, `\(F\)`, `\(D^*\)`, and `\(F^*\)`.

`\begin{align*}
D = \frac{\theta_\pi - \epsilon_1}{\sqrt{Var(\theta_\pi - \epsilon_1)}}
\end{align*}`

`\begin{align*}
F = \frac{\theta_W - \epsilon_1}{\sqrt{Var(\theta_W - \epsilon_1)}}
\end{align*}`

`\begin{align*}
D^* = \frac{\theta_\pi - \eta_1}{\sqrt{Var(\theta_\pi - \eta_1)}}
\end{align*}`

`\begin{align*}
F^* = \frac{\theta_W - \eta_1}{\sqrt{Var(\theta_W - \eta_1)}}
\end{align*}`

---

# Tajima's D and related tests

Tajima (1989) constructed the first test to detect difference between the SFS. 

His statistic, `\(D\)`, was defined as:

`\begin{align*}
D = \frac{\theta_\pi - \theta_W}{\sqrt{Var(\theta_\pi - \theta_W)}}
\end{align*}`

Originally designed to fit a normal distribution, however, none of these test statistics fit a parametric distribution very well.

### Calculation

- Only variable sites at each locus are needed
- The number of invariant sites do not figure into any calculations.

---

# Interpreting values of the test statistics

Tajima's `\(D\)`, Fu and Li's `\(D, F, D^*, F^*\)`:

`\begin{align*}
D = \frac{\theta_\pi - \theta_W}{\sqrt{Var(\theta_\pi - \theta_W)}}
\end{align*}`

After a sweep, all SNPs are low in frequency, `\(\theta_\pi\)` will be much lower than expected.

While statistics based on counts of segregating sites (like `\(\theta_W\)`) will be much closer to their expected values.

--

------

- All __negative__ when there has been a sweep


---

# Interpreting values of the test statistics

Tajima's `\(D\)`, Fu and Li's `\(D, F, D^*, F^*\)`:

`\begin{align*}
D = \frac{\theta_\pi - \theta_W}{\sqrt{Var(\theta_\pi - \theta_W)}}
\end{align*}`

Balancing selection lead to __an excess of intermediate frequency neutral variation__ surrounding a selected site.

In such case, `\(\theta_\pi\)` will be greater than `\(\theta_W\)` and other statistics.

--

------

- All __negative__ when there has been a sweep

- All __positive__ when there is balancing selection

--

- Are usually __significant__ when the values  `\(&gt; +2\)` or `\(&lt; -2\)`
  - The exact thresholds depend on sample size, number of SNPs, etc.


---
# The power of the SFS

The time window for positive selection is limited.
- #### Too early during the sweep 
  - Signal will be not strong enough
  
--
  
- #### Too late after the sweep
  - Both levels and frequencies of variants will have returned to normal

--

Power also determined by the distance between our studied loci and the location of the selected site.
- Because of the effect of the recombination.
- Move far away enough and there will be no signal of selection at all.


---

# The Hudson, Kreitman, Aguadé (HKA) Test

#### H0: If two loci evolve neutrally, they should have a similar ratio of polymorphism (within species) to divergence (between species)

--

Teases apart what is responsible for `\(\theta = 4N_e \mu\)`, testing for selection

Under a neutral model:

- All loci share the same `\(N_e\)`
  
- Neutral mutation rate `\(\mu\)` varies for different loci, but should be constant for the same locus in different species

--
  - High neutral mutation rate = high polymorphism and greater divergence between species
  
  -  Low rate = low polymorphism and less divergence
  
---

# The Hudson, Kreitman, Aguadé (HKA) Test

#### H0: If two loci evolve neutrally, they should have a similar ratio of polymorphism (within species) to divergence (between species)


To determine level of divergence
- Examine fixed differences between species
- Then look at number of polymorphisms between species

--

### The _Adh_ locus example

|  | _Adh_   | Control locus    | 
| :-------: | : ------ : | :-------: | 
| Polymorphism within species (S)  | 0.101    | 0.022   | 
| Divergence between species (D)  | 0.056     | 0.052   | 
| Ratio S/D (within/between)    | 1.80    | 0.42    | 
| `\(\chi^2\)` P-value | 0.016 |     |     |

--

Conclusion: Excess polymorphism at _Adh_ may be due to balancing selection, but HKA can’t determine the true cause


---

# Cautions and Prospects

- A review paper: [Detecting Natural Selection in Genomic Data](https://www.annualreviews.org/doi/abs/10.1146/annurev-genet-111212-133526) by Vitti, Grossman, and Sabeti, 2013.

  - Frequency-based methods (i.e., Tajima's D, Fay &amp; Wu's H, Ewens-Watterson test)
  - Gene-based methods (i.e., Ka/Ks, MKT)
  - Mutation rate-based methods (i.e., HKA test)

--

Rejection of the null hypothesis (H0: neutral theory) is not necessarily the same as demonstrating natural selection!
Because we have strong assumptions:
1. Constant Ne
2. Populations are at mutation-drift equilibrium
3. No gene flow
4. No recombination
...

--

Most of these tests are weak! They have little power to detect deviation unless these deviations are large.






    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
