<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Jinliang Yang" />


<title>Chapter 7: Values and Means</title>
<!-- Material Design fonts -->
<link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700">
<link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/icon?family=Material+Icons">
<script src="chapter7_lab_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="chapter7_lab_files/bootstrap-3.3.7/css/bootstrap.min.css" rel="stylesheet" />
<script src="chapter7_lab_files/bootstrap-3.3.7/js/bootstrap.min.js"></script>
<script src="chapter7_lab_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<script src="chapter7_lab_files/navigation-1.1/tabsets.js"></script>
<script src="chapter7_lab_files/navigation-1.1/codefolding.js"></script>
<link href="chapter7_lab_files/magnific-popup-1.1.0/magnific-popup.css" rel="stylesheet" />
<script src="chapter7_lab_files/magnific-popup-1.1.0/jquery.magnific-popup.min.js"></script>
<link href="chapter7_lab_files/bootstrap_material-0.1/bootstrap-material-design.min.css" rel="stylesheet" />
<link href="chapter7_lab_files/bootstrap_material-0.1/ripples.min.css" rel="stylesheet" />
<script src="chapter7_lab_files/bootstrap_material-0.1/material.min.js"></script>
<script src="chapter7_lab_files/bootstrap_material-0.1/ripples.min.js"></script>
<link href="chapter7_lab_files/material-0.1/material.css" rel="stylesheet" />
<script src="chapter7_lab_files/material-0.1/material.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span. { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */
</style>


<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("toc");
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

</head>

<body>

<div class="header-panel shadow z-2">
    <div class="container-fluid">
        <div class="row">
            <div class="col-xs-3">
        <div id="header">
    <h1 class="title">Chapter 7: Values and Means</h1>
                <h4 class="author">Jinliang Yang</h4>
                <h4 class="date">Sept.Â 06, 2018</h4>
        </div>
    </div>
</div>
</div>
</div>


<div class="container-fluid main-container">
    <div class="row">
      <nav class="col-xs-3 menu">
        <div id="toc">
        <ul>
        <li><a href="#trait-is-mostly-normal">Trait is mostly normal</a></li>
        <li><a href="#rice-data">Rice data</a></li>
        <li><a href="#population-structure">Population structure</a></li>
        <li><a href="#quality-control">Quality control</a></li>
        <li><a href="#genome-wide-association-studies">Genome-wide association studies</a></li>
        </ul>
        </div>
        
        
        
      </nav>
     <div class="pages col-xs-9">
     <div class="row">
       <div class="col-xs-10">



<div id="trait-is-mostly-normal" class="section level1">
<h1>Trait is mostly normal</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co">#pheno &lt;- sum(x)</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3">x &lt;-<span class="st"> </span><span class="kw">rnorm</span>(<span class="dt">n=</span><span class="dv">10000</span>, <span class="dv">0</span>, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">hist</span>(x)</a></code></pre></div>
<p><img src="chapter7_lab_files/figure-html/unnamed-chunk-1-1.png" width="576" /></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">x &lt;-<span class="st"> </span><span class="kw">rbeta</span>(<span class="dt">n=</span><span class="dv">10000</span>, <span class="fl">0.5</span>, <span class="fl">0.5</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">hist</span>(x)</a></code></pre></div>
<p><img src="chapter7_lab_files/figure-html/unnamed-chunk-1-2.png" width="576" /></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">x &lt;-<span class="st"> </span><span class="kw">runif</span>(<span class="dt">n=</span><span class="dv">10000</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw">hist</span>(x)</a></code></pre></div>
<p><img src="chapter7_lab_files/figure-html/unnamed-chunk-1-3.png" width="576" /></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">out &lt;-<span class="st"> </span><span class="kw">c</span>()</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10000</span>){</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">  p &lt;-<span class="st"> </span><span class="kw">rbinom</span>(<span class="dt">n=</span><span class="dv">1000</span>, <span class="dt">size=</span><span class="dv">1</span>, <span class="dt">prob=</span><span class="fl">0.9</span>)</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">  </a>
<a class="sourceLine" id="cb4-5" data-line-number="5">  phi &lt;-<span class="st"> </span><span class="kw">sum</span>(x<span class="op">*</span>p)</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">  out &lt;-<span class="st"> </span><span class="kw">c</span>(out, phi)</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">}</a>
<a class="sourceLine" id="cb4-8" data-line-number="8"></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="kw">hist</span>(out, <span class="dt">breaks=</span><span class="dv">30</span>)</a></code></pre></div>
<p><img src="chapter7_lab_files/figure-html/unnamed-chunk-1-4.png" width="576" /></p>
</div>
<div id="rice-data" class="section level1">
<h1>Rice data</h1>
<p>Download the Rice Diversity Panel data <code>RiceDiversity.44K.MSU6.Genotypes_PLINK.zip</code> from <a href="http://ricediversity.org/data/sets/44kgwas/">http://ricediversity.org/data/sets/44kgwas/</a>.</p>
<div id="genotype-data" class="section level2">
<h2>Genotype data</h2>
<p>The genotype data set is provided in the PLINK format. We will use the <code>read_ped</code> function from the <code>BGLR</code> package to read the PLINK format files into the R environment.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># install.packages(&quot;BGLR&quot;)</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw">library</span>(BGLR)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">out&lt;-<span class="st"> </span><span class="kw">read_ped</span>(<span class="st">&quot;../../STAT892/data/Genotypes/sativas413.ped&quot;</span>)</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">p=out<span class="op">$</span>p</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">n=out<span class="op">$</span>n</a>
<a class="sourceLine" id="cb5-6" data-line-number="6">out=out<span class="op">$</span>x</a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#Recode snp to 0,1,2 format using allele 1</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co"># 0 --&gt; 0</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co"># 1 --&gt; 1</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co"># 2 --&gt; NA</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co"># 3 --&gt; 2</span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12">out[out<span class="op">==</span><span class="dv">2</span>]=<span class="ot">NA</span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13">out[out<span class="op">==</span><span class="dv">3</span>]=<span class="dv">2</span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14">W &lt;-<span class="st"> </span><span class="kw">matrix</span>(out, <span class="dt">nrow=</span>p, <span class="dt">ncol=</span>n, <span class="dt">byrow=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb5-15" data-line-number="15">W &lt;-<span class="st"> </span><span class="kw">t</span>(W) </a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="kw">dim</span>(W) <span class="co"># # 413 x 36901</span></a></code></pre></div>
</div>
<div id="accession-ids" class="section level2">
<h2>Accession IDs</h2>
<p>The accession IDs (individual IDs) are included in <code>.fam</code> file.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># accession ID</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">fam &lt;-<span class="kw">read.table</span>(<span class="st">&quot;../../STAT892/data/Genotypes/sativas413.fam&quot;</span>, <span class="dt">header =</span> <span class="ot">FALSE</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)  </a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="kw">head</span>(fam)</a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="kw">rownames</span>(W) &lt;-<span class="st"> </span>fam<span class="op">$</span>V2 <span class="co"># 413 x 36901</span></a></code></pre></div>
</div>
<div id="phenotype-data" class="section level2">
<h2>Phenotype data</h2>
<p>We will use the <code>read.table</code> function to read the phenotype file <code>RiceDiversity_44K_Phenotypes_34traits_PLINK.txt</code> from <a href="http://www.ricediversity.org/data/sets/44kgwas/RiceDiversity_44K_Phenotypes_34traits_PLINK.txt">here</a>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># phenotypes</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2">rice.pheno &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;http://www.ricediversity.org/data/sets/44kgwas/RiceDiversity_44K_Phenotypes_34traits_PLINK.txt&quot;</span>, <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="kw">table</span>(<span class="kw">rownames</span>(W) <span class="op">==</span><span class="st"> </span>rice.pheno<span class="op">$</span>NSFTVID)</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">y &lt;-<span class="st"> </span><span class="kw">matrix</span>(rice.pheno<span class="op">$</span>Flowering.time.at.Arkansas) <span class="co"># # use the first trait </span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="kw">rownames</span>(y) &lt;-<span class="st"> </span>rice.pheno<span class="op">$</span>NSFTVID</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">index &lt;-<span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(y)</a>
<a class="sourceLine" id="cb7-7" data-line-number="7">y &lt;-<span class="st"> </span>y[index, <span class="dv">1</span>, drop=<span class="ot">FALSE</span>] <span class="co"># 374</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8">W &lt;-<span class="st"> </span>W[index, ] <span class="co"># 374 x 36901</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="kw">table</span>(<span class="kw">rownames</span>(W) <span class="op">==</span><span class="st"> </span><span class="kw">rownames</span>(y))</a></code></pre></div>
</div>
</div>
<div id="population-structure" class="section level1">
<h1>Population structure</h1>
<p>This data set is known to exhibit a subpopulation structure as the panel contains indica, aus, temperate japonica, tropical japonica, aromatic, and highly admixed accessions. We will create a PC plot to visualize the extent of population structure in rice.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># PC plots</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2">gp &lt;-<span class="kw">read.csv</span>(<span class="st">&quot;http://ricediversity.org/data/sets/44kgwas/RiceDiversity.44K.germplasm.csv&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">skip =</span> <span class="dv">1</span>,  <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)   <span class="co"># 431 x 12</span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3">gp2 &lt;-<span class="st"> </span>gp[<span class="kw">match</span>(<span class="kw">rownames</span>(y), gp<span class="op">$</span>NSFTV.ID), ]</a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="kw">table</span>(<span class="kw">rownames</span>(y) <span class="op">==</span><span class="st"> </span>gp2<span class="op">$</span>NSFTV.ID)</a>
<a class="sourceLine" id="cb8-5" data-line-number="5"></a>
<a class="sourceLine" id="cb8-6" data-line-number="6">    </a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="kw">plot</span>(gp2<span class="op">$</span>PC1, gp2<span class="op">$</span>PC2, <span class="dt">xlab=</span><span class="st">&quot;PC1&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;PC2&quot;</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>)[<span class="kw">factor</span>(gp2<span class="op">$</span>Sub.population)])</a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="kw">legend</span>(<span class="dt">x=</span><span class="st">&quot;topleft&quot;</span>, <span class="dt">legend =</span> <span class="kw">levels</span>(<span class="kw">factor</span>(gp2<span class="op">$</span>Sub.population)), <span class="dt">col=</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>), <span class="dt">pch=</span><span class="dv">1</span>, <span class="dt">cex=</span><span class="fl">0.6</span>)</a></code></pre></div>
</div>
<div id="quality-control" class="section level1">
<h1>Quality control</h1>
<div id="genotype-imputation" class="section level2">
<h2>Genotype imputation</h2>
<p>Replace missing marker genotypes with mean values. Then store the marker genotypes in a matrix object <code>W</code>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(W)){</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">  W[,j] &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">is.na</span>(W[,j]), <span class="kw">mean</span>(W[,j], <span class="dt">na.rm=</span><span class="ot">TRUE</span>), W[,j])</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">}</a></code></pre></div>
</div>
<div id="compute-allele-frequencies-for-all-snps" class="section level2">
<h2>Compute allele frequencies for all SNPs</h2>
<p>Recall that allele frequency of <em>A</em> is given by <span class="math display">\[ 
f(A) = p = \frac{2 \times (\text{no. of } AA \text{ individuals}) + 1 \times (\text{no. of } Aa \text{ individuals})}{2 \times \text{total no. of individuals}}.
\]</span> We can rewrite this equation into <span class="math display">\[ 
f(A) = p = \frac{(\text{no. of } A \text{ allele in the population})}{2 \times \text{total no. of individuals}}.
\]</span> This suggests that all we need is the number of <span class="math inline">\(A\)</span> allele or reference allele <span class="math inline">\(a\)</span> for each SNP. The <code>sum</code> function returns the number of reference allele <span class="math inline">\(A\)</span>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw">sum</span>(W[,<span class="dv">1</span>]) <span class="co"># sum of A allele in the first SNP</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="kw">sum</span>(W[,<span class="dv">2</span>]) <span class="co"># sum of A allele in the second SNP</span></a></code></pre></div>
<p>How to repeat this operation for 3,6901 SNPs? The <code>colSums</code> function returns the sum of each column in a matrix as a vector.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">colSums</span>(W) </a></code></pre></div>
<p>Note that <code>colSums(W)</code> gives the numerator of the above equation. We then divide this number by <span class="math inline">\(2 \times \text{total no. of individuals}\)</span>. The function <code>nrows</code> returns the number of rows.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">p &lt;-<span class="st"> </span><span class="kw">colSums</span>(W) <span class="op">/</span><span class="st"> </span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span><span class="kw">nrow</span>(W)) <span class="co"># or colMean(X) / 2</span></a></code></pre></div>
<p>The variable <code>p</code> is a vector and it contains the allele frequencies of reference allele.</p>
</div>
<div id="minor-allele-frequency-maf" class="section level2">
<h2>Minor allele frequency (MAF)</h2>
<p>Perform a quality control analysis by removing markers with MAF &lt; 0.05. How many markers are removed? Save the filtered genotype matrix in <code>W2</code>.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">maf &lt;-<span class="st"> </span><span class="kw">ifelse</span>(p <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>, <span class="dv">1</span><span class="op">-</span>p, p)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">maf.index &lt;-<span class="st"> </span><span class="kw">which</span>(maf <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">W2 &lt;-<span class="st"> </span>W[, <span class="op">-</span>maf.index]  <span class="co"># 374 x 33558 </span></a></code></pre></div>
</div>
</div>
<div id="genome-wide-association-studies" class="section level1">
<h1>Genome-wide association studies</h1>
<p>Fit a single-marker-based linear mixed model by using the <code>GWAS</code> function in the <a href="https://cran.r-project.org/web/packages/rrBLUP/index.html">rrBLUP</a> R package. Report the -log10 of p-values for SNP effects.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="co"># install.packages(&quot;rrBLUP&quot;)</span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="kw">library</span>(rrBLUP)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">map &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;../../STAT892/data/Genotypes/sativas413.map&quot;</span>, <span class="dt">header =</span> <span class="ot">FALSE</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb14-4" data-line-number="4">my.geno &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">marker=</span>map[,<span class="dv">2</span>], <span class="dt">chrom=</span>map[,<span class="dv">1</span>], <span class="dt">pos=</span>map[,<span class="dv">4</span>], <span class="kw">t</span>(W<span class="dv">-1</span>), <span class="dt">check.names =</span> <span class="ot">FALSE</span>) <span class="co"># W = \in{-1, 0, 1}</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5">my.pheno &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">NSFTV_ID=</span><span class="kw">rownames</span>(y), <span class="dt">y=</span>y) </a>
<a class="sourceLine" id="cb14-6" data-line-number="6"></a>
<a class="sourceLine" id="cb14-7" data-line-number="7">rel &lt;-<span class="st"> </span><span class="kw">GWAS</span>(my.pheno, my.geno, <span class="dt">min.MAF=</span><span class="fl">0.05</span>, <span class="dt">P3D=</span><span class="ot">TRUE</span>, <span class="dt">plot=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="kw">head</span>(rel<span class="op">$</span>y)</a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="kw">tail</span>(rel<span class="op">$</span>y)</a></code></pre></div>
<div id="manhattan-plot" class="section level2">
<h2>Manhattan plot</h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="co"># install.packages(&quot;qqman&quot;)</span></a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="kw">library</span>(qqman)</a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="kw">manhattan</span>(<span class="dt">x =</span> rel, <span class="dt">chr =</span> <span class="st">&quot;chrom&quot;</span>, <span class="dt">bp =</span> <span class="st">&quot;pos&quot;</span>, <span class="dt">p =</span> <span class="st">&quot;y&quot;</span>, <span class="dt">snp =</span> <span class="st">&quot;marker&quot;</span>, <span class="dt">col =</span> <span class="kw">c</span>(<span class="st">&quot;blue4&quot;</span>, <span class="st">&quot;orange3&quot;</span>), <span class="dt">logp =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
</div>
</div>

    </div>
    <div class="col-xs-2">
        <div class="btn-toolbar">
    <div class="btn-group">
    <a href="#" class="btn btn-raised btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span class="text-muted">Code</span> <span class="text-muted caret"></span></a>
    <ul class="dropdown-menu" style="min-width: 50px;">
    <li><a id="rmd-show-all-code" href="#">Show All</a></li>
    <li><a id="rmd-hide-all-code" href="#">Hide All</a></li>
    </ul>
    </div>
    </div>
        </div>
  </div>
  </div>
  </div>
  <div class="row">
    </div>
  </div>

<script>
$(document).ready(function () {
  // add bootstrap table styles to pandoc tables
  $('tr.header').parent('thead').parent('table').addClass('table table-striped table-hover');

    var images = $('.pages img');
  images.filter(function() {
      if ($(this).parent().attr("class") == "figure") {
          return(false)
      } else {
          return(true);
      }
  }).wrap("<div class='figure'></div>");
  images.addClass("image-thumb").wrap("<div class='panel-body'></div>");
  $('.figure p.caption').wrap("<div class='panel-footer'></div>");
  $('.figure').addClass('panel panel-default');
  
    $('.pages img')
 	  .addClass("image-lb");
  $('.pages').magnificPopup({
	      type:'image',
	      closeOnContentClick: false,
	      closeBtnInside: false,
        delegate: 'img',
	      gallery: {enabled: false },
          removalDelay: 500,
          callbacks: {
              beforeOpen: function() {
                // just a hack that adds mfp-anim class to markup
                this.st.image.markup = this.st.image.markup.replace('mfp-figure', 'mfp-figure mfp-with-anim');
              }
          },
          mainClass: 'mfp-move-from-top',
	      image: {
	        verticalFit: true,
            titleSrc: 'alt'
	      }
 	    });
 	
    
    $('#toc ul li').first().addClass("active");
    $('#toc ul li').attr("data-target", function() {
        return($(this).children("a").attr("href"));
    })
    $('body .section.level1').first().addClass("active");
    $('body .section.level1').not('.active').hide();
    
    $('#toc a[href*="#"]').click(function() {

      var id = $(this).attr("href");
      if (id === "#") return;
      if (id.substring(0, 8) === "#dyntab-") return;
      toggle_page(id);

      // Menu
      var menu_entry = $(".menu li[data-target='"+id+"']");
      menu_entry.addClass("active");
      $(".menu li").not(menu_entry).removeClass("active"); 
      

    });

    function toggle_page(id) {
      $(".page").not(page).removeClass("active").hide();
      window.page = id;
      var page = $(window.page);
      window.location.hash = window.page;
      //$(this).addClass("active");

      page.show();

      var totop = setInterval(function () {
        $(".pages").animate({scrollTop: 0}, 0);
      }, 10);

      setTimeout(function () {
        page.addClass("active");
        setTimeout(function () {
          clearInterval(totop);
        }, 1000);
      }, 100);

      window.dispatchEvent(new Event('resize'));

    }


    $(".menu li").click(function () {

      toggle_page($(this).data("target"));

      // Menu
      if (!$(this).data("target")) return;
      if ($(this).is(".active")) return;
      $(".menu li").not($(this)).removeClass("active");
      $(this).addClass("active");

    });
  
    


    window.page = window.location.hash;
    if (window.page != "") {
      $(".menu").find("li[data-target=" + window.page + "]").trigger("click");
    }

    /* init material bootstrap js */
    $.material.init();
});
</script>




<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
