---
title: "Breeding value interpretation"
author: "Jinliang Yang"
date: "March 25, 2024"
output:
  xaringan::moon_reader:
    css: ["default", "../../asset/unl-150.css"]
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
---

```{r setup, include=FALSE}
options(htmltools.dir.version = FALSE)
```

# Breeding value and dominance deviation


| Genotype  |  Value as deviated from $M$      | Breeding Value  | Dominance Deviation  | 
| :-------: | :-------: | :-----------: | :-----------: | :-------: | :-------: | 
| $A_1A_1$  |   $2q(\alpha - qd)$   | $2q\alpha$    |   $-2q^2d$   |  
| $A_1A_2$  |   $(q-p)\alpha + 2pqd$  | $(q-p)\alpha$ |    $2pqd$   |  
| $A_2A_2$  |   $-2p(\alpha + pd)$  | $-2p\alpha$   |  $-2p^2d$   |   

--
### G = A + D

- The mean of the BV is **zero**

- It follows that the mean dominance deviation is **zero**.



---
# A Linear Regression Perspective

### G = A + D

- **A** repsents the breeding value (i.e., A = $\alpha_i + \alpha_j$)
- **D** represents the dominance deviation

--

### Further breakdown __A__:

\begin{align*}
 G = & \alpha_1N_1 + \alpha_2N_2 + \delta
\end{align*}

--

- $\alpha_i$ is the average effect of allele $i$ and $\alpha = \alpha_1 - \alpha_2$
- $N_i$ is the number of allele $i$ carried by the genotype
- $N \in \{0, 1, 2\}$ for a bi-allelic locus and $N_1 + N_2 = 2$ 

--

Therefore,

\begin{align*}
 G = & \alpha_1N_1 + \alpha_2N_2 + \delta =  \alpha_1N_1 + \alpha_2(2 - N_1) + \delta  \\
   = & 2\alpha_2 + (\alpha_1 - \alpha_2)N_1 + \delta \\
   = & (2\alpha_2 + \delta) + \alpha N_1
\end{align*}



---
# Graphical Representation


| Genotype  |  Value as deviated from $M$      | Breeding Value  | Dominance Deviation  | 
| :-------: | :-------: | :-----------: | :-----------: | :-------: | :-------: | 
| $A_1A_1$  |   $2q(\alpha - qd)$   | $2q\alpha$    |   $-2q^2d$   |  
| $A_1A_2$  |   $(q-p)\alpha + 2pqd$  | $(q-p)\alpha$ |    $2pqd$   |  
| $A_2A_2$  |   $-2p(\alpha + pd)$  | $-2p\alpha$   |  $-2p^2d$   |   

An R function to calculate genotypic values

```{r}
gfunction <- function(a=1, d=0, p=1/2){
   # a: additive effect
   # d: dominance effect
   # p: allele frequency for the A1 allele
   q = 1-p
   # allele sub effect
   alpha <- a + d*(q-p)
   a1a1 <- 2*alpha*q
   a1a2 <- (q-p)*alpha
   a2a2 <- -2*p*alpha
   
   # population mean
   M <- a*(p-q) + 2*d*p*q
   # return a data.frame with genotype values and breeding values
   return(data.frame(N1=c(0,1,2), gv=c(-a-M,d-M,a-M), bv=c(a2a2, a1a2, a1a1)))
}
```

---
# Graphical Representation

### Apply the R function

```{r}
out <- gfunction(a=1, d=-1, p=1/3)
#out
out$dd <- out$gv - out$bv
out
```

---
# Plot the results

```{r fig.align='center', out.width='40%'}
plot(out$N1, out$gv, xlab="Genotype", ylab="", ylim=c(-1, 2), cex.lab=1.5, xaxt="n", pch=17, cex=2, col="red"); 
# add the axis and labels
axis(1, at=c(0, 1, 2), labels=c("A2A2", "A1A2", "A1A1")); 
# add y-axis title on the right side
mtext("Breeding Value", side = 4, line = 1, cex=1.5, col="blue"); 
# add y-axis title on the left side
mtext("Genotypic Value", side = 2, line = 2, cex=1.5, col="red")
# add breeding values
points(out$N1, out$bv, cex=2, col="blue")
# join the points by a line
lines(out$N1, out$bv, lwd=2, col="blue")
```



---

# Graphical Representation

The relationship between genotypic values, breeding values and dominance deviations.

\begin{align*}
G = & 2\alpha_2  + \alpha N_1 + \delta
\end{align*}


--

.pull-left[
<div align="center">
<img src="linear.png" height=350>
</div>
]

--

.pull-right[
- The __slope__ is the average effect of allele substitution. 
  - As $A_1$ is substituted by $A_2$, the breeding value increaes at a rate equal to $\alpha$.

- Dominance can be seen as __residuals__ from the fitted regression line.
  - Dominance deviation are the differences between the genotypic values and breeding values.
  
]

---
# Predicting BV: a real-world data

```{r}
# read phenotype and genotype data
flowering <- read.table("https://jyanglab.com/slides/flowering.csv", 
                        sep=",", header=TRUE)
dim(flowering)

head(flowering)
table(flowering$type)
f <- subset(flowering, type == "training")
```

---
# Visualize the phenotypic data

```{r fig.align='center', message=FALSE, warning=FALSE, out.width='40%'}
library(ggplot2)
p <- ggplot(f, aes(x=Flowering_time, fill=as.factor(snp1))) +
  geom_histogram(position="dodge")+
  # Use brewer color palettes
  scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) +
  guides(fill=guide_legend(title="Genotype")) +
  theme_classic() +
  theme(legend.position=c(0.8, 0.8), axis.text=element_text(size=20),
              axis.title=element_text(size=20))
p
```

---
# A real-world data

Let's find out the breeding value for each individual.

--

###  Training

- #### Step1: For each SNP, find out $a, d, p$, and $M$

- #### Step2: Compute average effect for A1 and A2

--

###  Prediction

- #### Step3: Sum up the BV of all SNPs

---
#  Training

#### Step1: For each SNP, find out __ $a$__, __ $d$__, $p$, and $M$

The number of copies of the A1 allele:
- A1A1 -> 2 
- A1A2 -> 1
- A2A2 -> 0

```{r}
A1A1 <- mean(subset(f, snp1 == 2)$Flowering_time, na.rm=TRUE)
A1A2 <- mean(subset(f, snp1 == 1)$Flowering_time, na.rm=TRUE)
A2A2 <- mean(subset(f, snp1 == 0)$Flowering_time, na.rm=TRUE)
A1A1
A1A2
A2A2
```

---

#  Training

#### Step1: For each SNP, find out __ $a$__, __ $d$__, $p$, and $M$

```{r}
### mid point
m <- (A1A1 + A2A2)/2
a <- A1A1 - m
d <- A1A2 - m
```

---
#  Training

#### Step1: For each SNP, find out $a, d$, __ $p$__, and $M$


Allele freq for __A1__ (Note it is in __`2`__ coding)

```{r}
table(f$snp1)
```

--

```{r}
getp <- function(df=f$snp1){
  t <- as.data.frame(table(df))
  c11 <- subset(t, df==2)$Freq
  c12 <- subset(t, df==1)$Freq
  c22 <- subset(t, df==0)$Freq
  return((2*c11+c12)/(2*c11+2*c12 +2*c22))
}
p <- getp(df=f$snp1)
p
q <- 1 -p
```

---
# Get population mean

#### Step1: For each SNP, find out $a, d$, $p$, and __ $M$__

```{r}
M1 <- mean(f$Flowering_time, na.rm=T)
M1

M2 <- a*(p-q) + 2*d*p*q
M2
M1 - M2

m
```


---
#  Training

#### Step2: Average effect for A1 and A2

| Genotype  |  Value as deviated from $M$      | Breeding Value  | Dominance Deviation  | 
| :-------: | :-------: | :-----------: | :-----------: | :-------: | :-------: | 
| $A_1A_1$  |   $2q(\alpha - qd)$   | $2q\alpha$    |   $-2q^2d$   |  
| $A_1A_2$  |   $(q-p)\alpha + 2pqd$  | $(q-p)\alpha$ |    $2pqd$   |  
| $A_2A_2$  |   $-2p(\alpha + pd)$  | $-2p\alpha$   |  $-2p^2d$   |   

--

```{r}
# allele sub effect
alpha <- a + d*(q-p) 
a1a1 <- 2*alpha*q 
a1a2 <- (q-p)*alpha 
a2a2 <- -2*p*alpha 
```

--

Apply the previous `gfunction` to get BV for snp1

```{r}
out2 <- gfunction(a=a, d=d, p=p)
out2
```

---

### Repeat the process for snp2 and snp3

```{r}
bv <- function(f, snpid="snp1", trait="Flowering_time"){
  A1A1 <- mean(f[f[, snpid]==0, trait], na.rm=TRUE)
  A1A2 <- mean(f[f[, snpid]==1, trait], na.rm=TRUE)
  A2A2 <- mean(f[f[, snpid]==2, trait], na.rm=TRUE)
  
  m <- (A1A1 + A2A2)/2
  a <- A1A1 - m
  d <- A1A2 - m
  p <- getp(df=f[, snpid])
  q <- 1 -p
  
  alpha <- a + d*(q-p) 
  a1a1 <- 2*alpha*q
  a1a2 <- (q-p)*alpha
  a2a2 <- -2*p*alpha
  M <- a*(p-q) + 2*d*p*q
  out <- data.frame(N1=c(0,1,2), gv=c(-a-M,d-M,a-M), bv=c(a2a2, a1a2, a1a1))
  return(out)
}

bv(f, snpid="snp1", trait="Flowering_time")
```

---

# Prediction

### Step3: Sum up the BV of all SNPs


BV is the sum of the average effects of the alleles an individual carries.

\begin{align*}
BV =  \sum_{i=1}^k\sum_{j=1}^2\alpha_{ij}
\end{align*}

Where summation occurs across the number of loci ( $k$ ) and the two alleles present at each locus.

--

```{r}
b1 <- bv(f, snpid="snp1", trait="Flowering_time")
b2 <- bv(f, snpid="snp2", trait="Flowering_time")
b3 <- bv(f, snpid="snp3", trait="Flowering_time")
b3
```

---

# Prediction

### Step3: Sum up the BV of all SNPs

```{r}
t <- subset(flowering, type=="test")
dim(t)
head(t, 4)
```

--

BV for individual `F356`:

```{r}
b1[b1$N1 ==2, ]$bv + b2[b2$N1 ==2, ]$bv + b3[b3$N1 ==2, ]$bv + M1
```

---

# Prediction

### Step3: Sum up the BV of all SNPs

```{r}
mygs <- function(t, b1, b2, b3){
  t$yhat <- -9
  for(i in 1:nrow(t)){
    t$yhat[i] <- b1[b1$N1 == t$snp1[i], ]$bv + b2[b2$N1 == t$snp2[i], ]$bv + b3[b3$N1 == t$snp3[i], ]$bv
  }
  return(t)
}
out <- mygs(t, b1, b2, b3)
out <- subset(out, !is.na(Flowering_time))
cor.test(out$Flowering_time, out$yhat+M1)
```




