<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Phenotypic values and variances</title>
    <meta charset="utf-8" />
    <meta name="author" content="Jinliang Yang" />
    <meta name="date" content="2022-03-24" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="asset/unl-150.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Phenotypic values and variances
### Jinliang Yang
### Mar.Â 24, 2022

---





# Select for shorter plants

&lt;div align="center"&gt;
&lt;img src="height.png" height=200&gt;
&lt;/div&gt;

- Tom collected a population of 20 maize landraces to select short plants.
- Then phenotypically characterized the 20 landraces and selected two of them as the founder lines to make an F2 population.
- After phenotyping and genotyping, Tom needs to determine which individual plants be more valuable to be selected and crossed.

--


```r
geno &lt;- read.table("data/geno.txt", header=FALSE)
dim(geno)
head(geno)
names(geno) &lt;- c("chr", "pos", "ref", "alt", paste0("plant", 1:20))
```


---

# Create an F2 population

- To make the F2 population, Tom chose `plant1` and `plant18` as the parents.
-  `plant1` and `plant18` were selfed five generations to be pure inbreds.


```r
### Just sample 100 markers from this Mt chr
set.seed(12579)
markers &lt;- sample(1:nrow(geno), size=100)
f &lt;- geno[sort(markers), c("chr", "pos", "ref", "alt", "plant1", "plant18")]
# select just one haplotype
f$plant1 &lt;- gsub("/.*", "", f$plant1)
f$plant18 &lt;- gsub("/.*", "", f$plant18)
# recoding to use -1,0,1
f[f==0] &lt;- -1
# simulate the recombination rate
f$cM &lt;- f$pos/5000
```


---

# Create an F2 population

- To make the F2 population, Tom chose `plant1` and `plant18` as the parents.
-  `plant1` and `plant18` were selfed five generations to be pure inbreds.


```r
install.packages("devtools")
library(devtools)
install_github("lian0090/simuPoisson")
library(simuPoisson)
```

--

Then, simulate an F2 population with 200 individuals and 100 SNP markers.


```r
pgeno &lt;- t(f[, c("plant1", "plant18")])
pgeno &lt;- apply(pgeno, 2, as.numeric)
f2 &lt;- simuPoisson(pgeno, f$chr, f$cM, 200)

f2 &lt;- as.data.frame(f2)
names(f2) &lt;- paste0(f$chr, "_", f$pos)
write.table(f2, "data/f2_geno.csv", sep=",", quote=FALSE)
```

---

# Compute genotype frq and allele frq


```r
f2 &lt;- read.csv("data/f2_geno.csv")
table(f2[,1])
```

Note that in this F2 population, the SNP coding is `-1, 0, 1` for `\(A_1A1\)`, `\(A_1A_2\)`, and `\(A_2A_2\)`.


```r
table(f2[,6])
```

---

# Frequencies

#### Observed allele frequency


```r
# For A1 allele
p &lt;- (52*2+92)/((52+92+56)*2)
# For A2 allele
q &lt;- (56*2+92)/((52+92+56)*2)
```

--

#### Observed genotype frequency


```r
# For A1A1 genotype
A1A1 &lt;- 52/(52+92+56)
# For A1A2 genotype
A1A2 &lt;- 92/(52+92+56)
# For A2A2 genotype
A2A2 &lt;- 56/(52+92+56)
```

---
# Frequencies

| Genotype  | Freq.   | Value   | 
| :-------: |: ------- :| :-------: | 
| `\(A_1A_1\)`  | `\(p^2\)`   | `\(+a\)`    | 
| `\(A_1A_2\)`  | `\(2pq\)`   | `\(d\)`     | 
| `\(A_2A_2\)`  | `\(q^2\)`   | `\(-a\)`    |

--

### Predicted genotype frequency


```r
p^2
2*p*q
q^2

chisq.test(rbind(c(A1A1, A1A2, A2A2), c(p^2, 2*p*q, q^2)))
```

---

# Phenotype

&lt;div align="center"&gt;
&lt;img src="height.png" height=200&gt;
&lt;/div&gt;

From this F2 population, Tom measured the plant height for each individual plant. 
  
Phenotype in a population can be characterized in terms of its __mean__ and __variance__.



---

# Phenotype


```r
pheno &lt;- read.csv("data/f2_pheno.csv")
hist(pheno$height, main="Plant Height", xlab="Value (inch)", breaks=20)
```

--

### Combine genotype and phenotype files


```r
gp &lt;- cbind(pheno, f2)
```

| Genotype  | Freq.   | Value   | 
| :-------: |: ------- :| :-------: | 
| `\(A_1A_1\)`  | `\(p^2\)`   | `\(+a\)`    | 
| `\(A_1A_2\)`  | `\(2pq\)`   | `\(d\)`     | 
| `\(A_2A_2\)`  | `\(q^2\)`   | `\(-a\)`    |


---

# Genotypic value

`\(P = G + E\)`

`\(G = A + D\)`


Let's find out `\(a\)` and `\(d\)` at a specific Marker `Mt_24242`:

--


```r
ggplot(gp, aes(x=as.factor(Mt_24242), y=height, color=as.factor(Mt_24242))) +
    geom_boxplot() +
    geom_jitter(color="black", size=1, alpha=0.9) +
    scale_color_manual(values=c("#E69F00", "#56B4E9", "#fe6f5e"))+
    labs(title="Mt_24242", y="Plant Height", x = "Genotype")+
    theme_classic() +
    guides(color=FALSE) +
    theme(plot.title = element_text(size=20, face = "bold"), 
          axis.text=element_text(size=16, face="bold"),
          strip.text.y = element_text(size = 16, face = "bold"),
          axis.title=element_text(size=18, face="bold"),
          )
```

---

# Genotypic value



Let's find out `\(a\)` and `\(d\)` at a specific Marker `Mt_24242`:


```r
u &lt;- mean(gp$height) # population mean
# A1A1
h1 &lt;- mean(subset(gp, Mt_24242 == -1)$height) 
# A1A2
h12 &lt;- mean(subset(gp, Mt_24242 == 0)$height)
# A2A2
h2 &lt;- mean(subset(gp, Mt_24242 == 1)$height)
```

--


```r
a &lt;- (h2 - h1)/2
midpoint &lt;- h1+a
d &lt;- h12 - midpoint
```

---

# Allele Substitution Effect

The average effect of A1 and A2:


```r
alpha &lt;- a + d*(q - p)
alpha1 &lt;- q*alpha
alpha2 &lt;- -p*alpha
```

--

### Breeding value


The __Breeding value__ associated with Marker `Mt_24242` is defined as: the sum of `\(\alpha_i\)` and `\(\alpha_j\)`.
- Breeding value is the value of an individual as a parent!

`\begin{align*}
BV_{ij} = \mu + \alpha_i + \alpha_j
\end{align*}`


```r
bv1 = u+alpha1 + alpha1
bv2 = u+alpha2 + alpha2
bv12 = u+alpha1 + alpha2
```

---

# Genotypic value and breeding value


```r
plot(c(0, 1, 2), c(h1, h12, h2), xlab="Genotype",ylab="", cex.lab=1.5, xaxt="n", pch=17, cex=2, col="red", ylim=c(60, 95)); 
axis(1, at=c(0, 1, 2), labels=c("A1A1", "A1A2", "A2A2")); 
mtext("Breeding Value", side = 4, line = 1, cex=1.5, col="blue"); 
mtext("Genotypic Value", side = 2, line = 2, cex=1.5, col="red")
points(c(0, 1, 2), c(bv2, bv12, bv1), cex=2, col="blue")
lines(c(0, 1, 2), c(bv2, bv12, bv1), lwd=2, col="blue")
```

---

# Additive and dominance variance

P = A + D + E

### Phenotypic variance


```r
Vp &lt;- var(gp$height)
```

---

# Additive genetic variance: `\(V_A\)`

These breeding values have a mean of zero, and their variance is the sum of the products of the genotype frequencies and the squared breeding values:

`\begin{align*}
V_A &amp; = p^2(2q\alpha)^2 + 2pq(q-p)^2\alpha^2 + q^2(-2p\alpha)^2 \\
           &amp; = 2pq\alpha^2(2pq + (q-p)^2 + 2pq) \\
           &amp; = 2pq\alpha^2(p+q)^2 \\
           &amp; = 2pq\alpha^2 \\
           &amp; = 2pq(a + d(q-p))^2 \\
\end{align*}`

--


```r
Va &lt;- 2*p*q*(a + d*(q - p))^2
```

---

# Dominance genetic variance: `\(V_D\)`

The variance due to dominance deviations is:
the sum of the products of the genotype frequencies and the squared dominance deviation values.

`\begin{align*}
V_D &amp; = p^2(-2q^2d)^2 + 2pq(2pqd)^2 + q^2(-2p^2d)^2 \\
           &amp; = 4p^2q^2d^2(q^2 + 2pq + p^2) \\
           &amp; = 4p^2q^2d^2 \\
           &amp; = (2pqd)^2 \\
\end{align*}`

--


```r
Vd &lt;- (2*p*q*d)^2
```

---

# Additive and dominance variance

`\(G = A + D\)`


```r
Vg &lt;- Va + Vd
```

--

### `\(H^2\)` and `\(h^2\)` due to this SNP marker


```r
h2 &lt;- Va/Vp
H2 &lt;- Vg/Vp
```





    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
