<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Genomic Selection in Practice</title>
    <meta charset="utf-8" />
    <meta name="author" content="Jinliang Yang" />
    <meta name="date" content="2022-04-14" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="asset/unl-150.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Genomic Selection in Practice
### Jinliang Yang
### April 24, 2025

---






# GBLUP vs. RR-BLUP

### GBLUP

Using genome-wide markers to quantify the amount of information extracted from relatives.

### RR-BLUP

Quantifying the effects of genome-wide markers from a population of related individuals.

--

----------

Theory has shown that GBLUP and RR-BLUP are equivalent:
- If the number of QTL is large
- No major QTL are present
- And the QTL are evenly distributed across the genome
&gt; Fernando, 1998; Habier et al., 2007; Goddard, 2009.

---

# RR-BLUP vs. QTL mapping

In RR-BLUP, usually, the number of markers ( `\(p\)` ) is greatly larger than observation ( `\(n\)` ), or `\(p &gt;&gt; n\)` 
  - a large p, small n problem


### Shrinkage

The marker effects with RR-BLUP are __shrinkage toward zero__.

- Modeling `\(V_{M_i}\)` as constant across all loci caused an overshrinkage of effects for markers close to major QTL (large effect QTL).

- And an undershrinkage of effects for markers close to minor QTL.
&gt; Meuwissen et al., 2001.

--

Furthermore, for a given QTL, several marker loci will capture its effect if using the RR-BLUP method.

---

# Framework of GS

&lt;div align="center"&gt;
&lt;img src="gs.png" height=250&gt;
&lt;/div&gt;

#### Model training
In the context of GS, the lines with both genotype and phenotype data constitute a __training population__. =&gt; `\(\mathbf{\hat{m}}\)`

#### Prediction
Predict the performance of the untested lines using their genotype data.
`\(\mathbf{\hat{y}} = \mathbf{1}\mu + \mathbf{Z} \mathbf{\hat{m}}\)`

---

# Framework of GS

In the model training process, a __cross-validation__ method will be used within the training population.

- __k-fold__ cross-validation
  - The training population is divided in `\(k\)` sets (i.e., 5-fold). 
  
- __delete-one__ (or __leave-one-out__) cross-validation
  - Use `\(n-1\)` to train the model.

--

### Prediction accuracy

The prediction accuracy (denoted by `\(r_{MG}\)`) is the correlation between the true genotypic value and the genotypic value predicted from marker effects.

---

# A real world example: Loblolly pine data

Loblolly pine (Pinus taeda) data
&gt;Resende Jr. et al. (2012) ([DOI: 10.1534/genetics.111.137026](http://dx.doi.org/10.1534/genetics.111.137026))


- __Mating Design__: 70 full-sib families and 951 individuals in total using _a circular mating design_
- __Phenotyping__: 17 traits with distinct heritabilities and genetic architectures
- __Genotyping__: with 4,853 SNPs using the SNP array method. 
  - The full dataset can be downloaded from the paper. 

---

# A real world example: Loblolly pine data

In this example, we will use the breeding values of crown width across the planting beds at age 6 (CWAC6). 


```r
# read phenotype and SNP files
pheno_file &lt;- "https://jyanglab.com/img/data/DATA_nassau_age6_CWAC.csv"
geno_file &lt;- "https://jyanglab.com/img/data/Snp_Data.csv"
pheno &lt;- read.csv(pheno_file, header=TRUE, stringsAsFactors = FALSE)
hist(pheno$Derregressed_BV, main="Crown width at Age 6", xlab="width")
```

![](w13lab_gs2_files/figure-html/unnamed-chunk-1-1.png)&lt;!-- --&gt;

---
# Loblolly pine data


### Remove missing phenotypes 

There are some accessions containing no phenotype. We need to remove these accessions first.


```r
na.index &lt;-  which(is.na(pheno$Derregressed_BV))
# length(na.index)
pheno &lt;- pheno[-na.index, ]
# Keep genotypes for these remaining lines
geno &lt;- geno[geno$Genotype %in% pheno$Genotype, ]

# phenotypes 
y &lt;- pheno$Derregressed_BV
y &lt;- matrix(y, ncol=1)

# markers 
geno &lt;- geno[,-1] # 861 x 4853
geno[geno == -9] &lt;- NA
```


---
# Genotype data: SNP quality control

In the `geno` matrix, row indicates individual, column indicates SNPs.

### Missingness and MAF


```r
geno &lt;- read.csv(geno_file, header=TRUE, stringsAsFactors = FALSE)
dim(geno)

# Keep genotypes for these remaining lines
geno &lt;- geno[geno$Genotype %in% pheno$Genotype, ]
# markers 
geno &lt;- geno[,-1] # 861 x 4853
geno[geno == -9] &lt;- NA

# missing rate
missing &lt;- apply(geno, 2, function(x){sum(is.na(x))/length(x)})
# minor allele frequency
maf &lt;- apply(geno, 2, function(x){
  frq &lt;- mean(x, na.rm=TRUE)/2 # 1 allele
  return(ifelse(frq &gt; 0.5, 1-frq, frq))
})
```

---
# Genotype data: SNP quality control

In the `geno` matrix, row indicates individual, column indicates SNPs.


#### Plot the results

```r
hist(missing, breaks=100, col="blue", xlab="SNP Missing rate")
hist(maf, breaks=100, col="blue", xlab="Minor Allele Freq")
```

---
# SNP quality control

Removing SNPs with high missing rate (missingness &gt; 0.2) and low MAF (MAF &lt; 0.05)

- Question: How many markers are removed?


```r
idx1 &lt;- which(missing &gt; 0.2) #154
idx2 &lt;- which(maf &lt; 0.05) #1647
idx &lt;- unique(c(idx1, idx2)) #1784

geno2 &lt;- geno[, -idx]
dim(geno2)
```


---
# SNP quality control

### Missing marker imputation

Replace missing marker genotypes with __mean values__. Then store the marker genotypes in a matrix object `Z`. 


```r
Z &lt;- matrix(0, ncol=ncol(geno2), nrow=nrow(geno2))
for (j in 1:ncol(geno2)){
  #cat("j = ", j, '\n')
  Z[,j] &lt;- ifelse(is.na(geno2[,j]), mean(geno2[,j], na.rm=TRUE), geno2[,j])
}
# sum(is.na(Z))
write.table(Z, "data/Z.txt", sep="\t", row.names = FALSE, 
            col.names=FALSE, quote=FALSE)
```

---

# Genomic relationship

### SNP Matrix standardization

Standardize the genotype matrix to have a mean of zero and variance of one. Save this matrix as `Zs`. 


```r
Zs &lt;- scale(Z, center = TRUE, scale = TRUE)
# dimensions 
n &lt;- nrow(Zs)
m &lt;- ncol(Zs)
```

---

# Genomic relationship

### Calcualte genomic relationship

- Compute the second genomic relationship matrix of [VanRaden (2008)](https://www.ncbi.nlm.nih.gov/pubmed/18946147) using the entire markers. 
- Then add a very small positive constant (e.g., 0.001) to the diagonal elements so that `G` matrix is invertible. 


```r
# Given matrices x and y as arguments, return a matrix cross-product. 
# This is formally equivalent to (but usually slightly faster than) 
# the call t(x) %*% y (crossprod) or x %*% t(y) (tcrossprod).
G &lt;- tcrossprod(Zs) / ncol(Zs)
# G &lt;- Zs %*% t(Zs) /ncol(Zs)
G &lt;- G + diag(n)*0.001
```

---

# Solve MME for GBLUP

Set up mixed model equations (MME) by fitting the model:

  `$$\mathbf{y = 1\mu + Zu + e}$$`
  
- where `\(\mu\)` is the intercept, 
- `\(\mathbf{Z}\)` is the incident matrix of individuals, 
- `\(\mathbf{u}\)` is the breeding value of the individuals, 
- and `\(\mathbf{e}\)` is the residual. 

Directly take the inverse of LHS to obtain the solutions for GBLUP. Report the estimates of intercept and additive genetic values. Use `\(\lambda = 1.35\)`. 


```r
lambda &lt;- 1.35 # fit$Ve / fit$Vm
Ginv &lt;- solve(G)
ones &lt;- matrix(1, ncol=1, nrow=n)
Z &lt;- diag(n)
# Given matrices x and y as arguments, return a matrix cross-product. 
#This is formally equivalent to (but usually slightly faster than) 
#the call t(x) %*% y (crossprod) or x %*% t(y) (tcrossprod).
LHS1 &lt;- cbind(crossprod(ones), crossprod(ones, Z)) 
LHS2 &lt;- cbind(crossprod(Z, ones), crossprod(Z) +  Ginv*lambda)
LHS &lt;- rbind(LHS1, LHS2)
RHS &lt;- rbind( crossprod(ones, y), crossprod(Z,y) )
sol &lt;- solve(LHS, RHS)
head(sol)
tail(sol)
```

---

# R package: `rrBLUP`

Fit GBLUP by using the `mixed.solve` function in the [rrBLUP](https://cran.r-project.org/web/packages/rrBLUP/index.html) R package. 

- Report the estimates of intercept and additive genetic values. 
- Do they agree with previous estimates? 
- Also, report the estimated genomic heritability and the ratio of variance components `\(\lambda = \frac{V_e}{V_A}\)`. 


```r
#install.packages("rrBLUP")
library(rrBLUP)
fit &lt;- mixed.solve(y = y, K=G)
# additive genetic variance
fit$Vu
# residual variance
fit$Ve
# intercept 
fit$beta
# additive genetic values
head(fit$u)
tail(fit$u)
# genomic h2
fit$Vu / (fit$Vu + fit$Ve)
# ratio of variance components 
fit$Ve / fit$Vu
# plot(x=sol[-1], y=fit$u)
```

---

# RR-BLUP

Set up mixed model equations (MME) by fitting the model `\(\mathbf{y = 1b + Zm + e}\)`, where `\(\mathbf{b}\)` is the intercept, `\(\mathbf{Z}\)` is the standardized marker genotypes (`Zs`), `\(\mathbf{m}\)` is the additive marker genetic effects, and `\(\mathbf{e}\)` is the residual. 

`\begin{align*}
  \begin{bmatrix}
    \mathbf{\hat{b}} \\
    \mathbf{\hat{m}} \\
  \end{bmatrix}
  =
  \begin{bmatrix}
    \mathbf{X^{'}R^{-1}X} &amp; \mathbf{X^{'}R^{-1}Z} \\
    \mathbf{Z^{'}R^{-1}X} &amp; \mathbf{Z^{'}R^{-1}Z} + \mathbf{I} V_e/V_{M_i} \\
  \end{bmatrix}^{-1}
  \begin{bmatrix}
    \mathbf{X^{'}R^{-1}y} \\
    \mathbf{Z^{'}R^{-1}y} \\
  \end{bmatrix}
\end{align*}`

Directly take the inverse of LHS to obtain the solutions for marker-based GBLUP (RR-BLUP). Report the estimates of intercept and marker additive genetic effects. Use `\(\lambda = 4326.212\)`. 

--


```r
lambda &lt;- 4326.212 # fit$Ve / fit$Vu
ones &lt;- matrix(1, ncol=1, nrow=n)
I &lt;- diag(m)
LHS1 &lt;- cbind(crossprod(ones), crossprod(ones, Zs)) 
LHS2 &lt;- cbind(crossprod(Zs, ones), crossprod(Zs) +  I*lambda)
LHS &lt;- rbind(LHS1, LHS2)
RHS &lt;- rbind( crossprod(ones, y), crossprod(Zs,y) )
sol2 &lt;- solve(LHS, RHS)
head(sol2)
tail(sol2)
```

---

# Use `rrBLUP` package

Fit RR-BLUP by using the `mixed.solve` function in the [rrBLUP](https://cran.r-project.org/web/packages/rrBLUP/index.html) R package. 

- Report the estimates of intercept and marker additive genetic effects. 
- o they agree with the estimates with the manual calculation? 
- Also, report the ratio of variance components `\(\lambda = \frac{V_e}{V_A}\)`. 


```r
library(rrBLUP)
fit2 &lt;- mixed.solve(y = y, Z=Zs)
# marker additive genetic variance
fit2$Vu
# residual variance
fit2$Ve
# intercept 
fit2$beta
# marker additive genetic effects
head(fit2$u)
tail(fit2$u)
# ratio of variance components 
fit2$Ve / fit2$Vu

# plot(x=sol2[-1], y=fit2$u)
```

---

# K-fold validation

Repeat GBLUP but treat the first 600 individuals as a training set and predict the additive genetic values of the remaining individuals in the testing set. 
- What is the predictive correlation in the testing set? Use `\(\lambda = 1.348411\)`. 


```r
n.trn &lt;- 600
n.tst &lt;- 261
y.trn &lt;- y[1:n.trn]
y.tst &lt;- y[n.trn+1:n.tst]
Zs.trn &lt;- Zs[1:n.trn,]
Zs.tst &lt;- Zs[n.trn+1:n.tst,]

Gtrn &lt;- tcrossprod(Zs.trn) / ncol(Zs.trn)
Gtrn &lt;- Gtrn + diag(n.trn)*0.001
Gtst.trn &lt;- tcrossprod(Zs.tst, Zs.trn) / ncol(Zs.tst)
#Gtrn &lt;- G[1:n.trn, 1:n.trn]
#Gtst.trn &lt;- G[n.trn+1:n.tst, 1:n.trn]

lambda &lt;- 1.348411 # fit$Ve / fit$Vu
Ginv.trn &lt;- solve(Gtrn)
ones &lt;- matrix(1, ncol=1, nrow=n.trn)
Z &lt;- diag(n.trn)
LHS1 &lt;- cbind(crossprod(ones), crossprod(ones, Z)) 
LHS2 &lt;- cbind(crossprod(Z, ones), crossprod(Z) +  Ginv.trn*lambda)
LHS &lt;- rbind(LHS1, LHS2)
RHS &lt;- rbind( crossprod(ones, y.trn), crossprod(Z,y.trn) )
sol.trn &lt;- solve(LHS, RHS)

# prediction
y.hat &lt;- Gtst.trn %*% Ginv.trn %*% matrix(sol.trn[c(2:(n.trn+1))])
cor(y.hat, y[(n.trn+1):n])
# plot(y.hat, y[(n.trn+1):n])
```

---

# K-fold validation

Repeat RR-BLUP but treat the first 600 individuals as a training set and predict the additive genetic values of the remaining individuals in the testing set. 
- What is the predictive correlation in the testing set? Use `\(\lambda = 4326.212\)`. 
- Also, compare this predictive correlation to the one from GBLUP. 


```r
Zs.trn &lt;- Zs[1:n.trn, ]
Zs.tst &lt;- Zs[n.trn+1:n.tst, ]
lambda &lt;- 4326.212 # fit$Ve / fit$Vu
ones &lt;- matrix(1, ncol=1, nrow=n.trn)
I &lt;- diag(m)
LHS1 &lt;- cbind(crossprod(ones), crossprod(ones, Zs.trn)) 
LHS2 &lt;- cbind(crossprod(Zs.trn, ones), crossprod(Zs.trn) +  I*lambda)
LHS &lt;- rbind(LHS1, LHS2)
RHS &lt;- rbind( crossprod(ones, y.trn), crossprod(Zs.trn, y.trn) )
sol.trn &lt;- solve(LHS, RHS)

# prediction
y.hat2 &lt;- Zs.tst %*% matrix(sol.trn[-1])
# cor(y.hat2, y[(n.trn+1):n])
# plot(y.hat2, y[(n.trn+1):n])
```





    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
