<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Population genomics software</title>
    <meta charset="utf-8" />
    <meta name="author" content="Jinliang Yang" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="asset/unl-150.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Population genomics software
### Jinliang Yang
### Mar.Â 13, 2025

---





# Population genomics

- NGS and diversity measurement
  - `\(\theta_\pi\)`: pairwise necleotide diversity
  - `\(\theta_W\)`: watterson's `\(\theta\)`, using total number of segregating sites
  - `\(\epsilon_1 = S_1\)`: the number of derived singletons in a sample.
  - `\(\eta_1\)`: based on all singletons in a sample.

--

- Population differentiation
  - `\(F_{ST}\)`
  
--

- Scan for direct selection
  - `\(d_N/d_S\)` or `\(\pi_N/\pi_S\)`

--

- Scan for linked selection
  - Tajima's D
  - Fu and Li's D, F, D\*, F\*

---

# Population genomics

- NGS and diversity measurement
  - `\(\theta_\pi\)`: pairwise necleotide diversity
  - `\(\theta_W\)`: watterson's `\(\theta\)`, using total number of segregating sites
  - `\(\epsilon_1 = S_1\)`: the number of derived singletons in a sample.
  - `\(\eta_1\)`: based on all singletons in a sample.

### [ANGSD](http://www.popgen.dk/angsd/index.php/ANGSD)

ANGSD is a software for conducting population genomics analysis using next generation sequencing data.

One advantages of this software is that it can handle mapped reads to imputed genotype probabilities.

---

# Installation, however, is painful

### [Installation](http://www.popgen.dk/angsd/index.php/Installation)


```bash
cd $HOME/bin
```

### Install from github


```bash
git clone --recursive https://github.com/samtools/htslib.git
git clone https://github.com/ANGSD/angsd.git 
cd htslib;make;cd ../angsd ;make HTSSRC=../htslib
```

### Install by directly downloading


```bash
wget http://popgen.dk/software/download/angsd/angsd0.936.tar.gz
tar xf angsd0.936.tar.gz
cd htslib;make;cd ..
cd angsd
make HTSSRC=../htslib
cd ..
```

---

# Instead, use `Conda`

`Conda` is a package and environment manager!
  - by far the __easiest way to handle installing__ most of the tools we want to use in bioinformatics.

It has been installed on the HCC.

To learn more about conda, read this [introduction](https://astrobiomike.github.io/unix/conda-intro).

---

# Making a new environment

The simplest way we can create a new conda environment is like so:


```bash
conda create -n mypopgen
```

#### Entering an environment


```bash
conda activate mypopgen
```

#### Installing packages

The first thing I usually do is just search in a web-browser for `conda install` plus whatever program I am looking for.


```bash
conda install -c bioconda angsd
```

#### Exiting an environment


```bash
conda deactivate
```

---

# NGS and diversity measurement

#### Use our simulated data from `lab5`


```bash
cd largedata;
mkdir lab7
cp /common/jyanglab/jyang21/courses/2022-agro932-lab/largedata/lab5/bamlist.txt lab7
cp /common/jyanglab/jyang21/courses/2022-agro932-lab/largedata/lab5/sorted_l* lab7
cp /common/jyanglab/jyang21/courses/2022-agro932-lab/largedata/lab5/Zea_mays.B73_RefGen_v4.dna.chromosome.Mt.fa* lab7
```

--------

#### Activate my conda environment


```bash
conda activate mypopgen
angsd
```


```bash
module load angsd
angsd
```

request a computer node

```bash
srun --qos=short --nodes=1 --licenses=common --ntasks=4 --mem 8G --time 2:00:00 --pty bash
```

---

# ANGSD for diversity meansurement


```bash
cd lab7
module load samtools
samtools faidx Zea_mays.B73_RefGen_v4.dna.chromosome.Mt.fa

# run ANGSD to calculated folded SFS
angsd -bam bamlist.txt -out output -doMajorMinor 1 -doMaf 1 -doSaf 2 -uniqueOnly 0 -anc Zea_mays.B73_RefGen_v4.dna.chromosome.Mt.fa -minMapQ 30 -minQ 20 -nInd 20 -baq 1 -ref Zea_mays.B73_RefGen_v4.dna.chromosome.Mt.fa -GL 1

# use realSFS to calculate sfs
realSFS output.saf.idx -fold 1 &gt; output.sfs

# try this version
/common/jyanglab/gxu6/software/angsd/misc/realSFS output.saf.idx -fold 1 &gt; output.sfs
```

#### Copy the result to `cache/` folder


```bash
## cp sfs to the cache/ folder
cp output.sfs ../../cache/
```

---

# Calculate the thetas

#### For each site


```bash
/common/jyanglab/gxu6/software/angsd/misc/realSFS saf2theta output.saf.idx -sfs output.sfs -outname output
```

The output from the above command are two files `output.thetas.gz` and `output.thetas.idx`. 
- A formal description of these files can be found in the `doc/formats.pdf` in the angsd package. 

#### For stepping window


```bash
/common/jyanglab/gxu6/software/angsd/misc/thetaStat do_stat output.thetas.idx -win 5000 -step 1000  -outnames output.theta.5k.gz
# Copy the result to `cache/` folder
cp output.theta.5k.gz.pestPG ../../cache/
```


```bash
git add --all
git commit -m "theta values"
git push
```

---

# Visualize the results

In the local computer, using `R`:

#### Barplot for SFS


```r
s &lt;- scan('cache/output.sfs')
s &lt;- s[-c(1,length(s))]
s &lt;- s/sum(s)
barplot(s,names=1:length(s), main='SFS')
```

#### Histgram distribution of the theta values


```r
theta &lt;- read.table("cache/output.theta.5k.gz.pestPG", header=TRUE)
hist(theta$tW, xlab="theta_w", main="disverity")
```

#### Scatter plot of the Tajima's D values



```r
plot(theta$WinCenter, theta$Tajima, xlab="Physical position", ylab="Tajima's D", col="#5f9ea0", pch=16)
```


---

# Fst using `vcftools`

### input data: 
  - variant call format (or the VCF/BCF file)
  - need to determine the populations



```bash
module load bcftools
# you must be in your lab7/ folder
cp /common/jyanglab/jyang21/courses/2022-agro932-lab/largedata/lab5/snps.bcf .
bcftools view snps.bcf | head -n 40
```

sorted_l10.bam	sorted_l11.bam	sorted_l12.bam	sorted_l13.bam	sorted_l14.bam	sorted_l15.bam	sorted_l16.bam	sorted_l17.bam	sorted_l18.bam	sorted_l19.bam	sorted_l1.bam	sorted_l20.bam	sorted_l2.bam	sorted_l3.bam	sorted_l4.bam	sorted_l5.bam	sorted_l6.bam	sorted_l7.bam	sorted_l8.bam	sorted_l9.bam


```bash
for ((i=1;i&lt;=10;i++)) ; do echo "sorted_l$i.bam" &gt;&gt; pop1.txt;  done
for ((i=11;i&lt;=20;i++)) ; do echo "sorted_l$i.bam" &gt;&gt; pop2.txt;  done
```

---

# Fst using `vcftools`

### Window based Fst


```bash
module load vcftools
vcftools --bcf snps.bcf --weir-fst-pop pop1.txt --weir-fst-pop pop2.txt --fst-window-size 10000 --fst-window-step 1000 --out win_1k
```

### Store the Weir Fst results


```bash
## cp Fst to the cache/ folder
cp win_1k.windowed.weir.fst ../../cache/
```


---

# XP-CLR approach for selection scan

### input data: 
  - variant call format (VCF file only)
  - need to determine the populations


```bash
module load xpclr/1.1
module load bcftools
bcftools convert snps.bcf -O v -o snp.vcf 

xpclr --input snp.vcf --out ./xpclr_res.txt  --format vcf --samplesA pop1.txt --samplesB pop2.txt --chr Mt --start 1 --stop 20000 --ld 0.7 --maxsnps 200 --minsnps 200 --size 10000 --step 5000
```


---

# XP-CLR approach for selection scan

### using slurm script


```bash
cd ../../
pwd
```

`/common/jyanglab/jyang21/courses/2022-agro932-lab`


```bash
#!/bin/bash -l
#SBATCH -D /common/jyanglab/jyang21/courses/2022-agro932-lab
#SBATCH -o /common/jyanglab/jyang21/courses/2022-agro932-lab/slurm-log/xpclr-stdout-%j.txt
#SBATCH -e /common/jyanglab/jyang21/courses/2022-agro932-lab/slurm-log/xpclr-stderr-%j.txt
#SBATCH -J xpclr
#SBATCH -t 10:00:00
#SBATCH --mail-user=your_email_address@gmail.com
#SBATCH --mail-type=END #email if ends
#SBATCH --mail-type=FAIL #email if fails
set -e
set -u

### your script here:
module load xpclr/1.1
xpclr --input snp.vcf --out ./xpclr_res.txt  --format vcf --samplesA pop1.txt --samplesB pop2.txt --chr Mt --start 1 --stop 200000 --ld 0.7 --maxsnps 200 --minsnps 200 --size 10000 --step 5000
```



```bash
vi slurm-script/xpclr.sh
i
# copy and paste the above code
```

---

# XP-CLR approach for selection scan

### using slurm script


```bash
sbatch --licenses=common --ntasks=2 --mem=10G slurm-script/my_theta.sh
## check your job status
squeue | grep "YOUR USER ID"
```

### Store the XP-CLR results


```bash
cd largedata/lab7
cp xpclr_res.txt ../../cache/
```

Type git command to version control your results


```bash
git add --all
git commit -m "Fst and XP-CLR results"
git push
```










    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
