<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Journal Article Review</title>
    <meta charset="utf-8" />
    <meta name="author" content="Jinliang Yang" />
    <meta name="date" content="2024-02-23" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="../../asset/unl-150.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# Journal Article Review
]
.author[
### Jinliang Yang
]
.date[
### Feb. 23, 2024
]

---






# Points to cover in the review:

- ### 1. Novelty

- ### 2. Quality

- ### 3. Clarity

- ### 4. Reproducibility

---

# Points to cover in the review:
### Novelty

- What are the main claims of the paper and how significant are they? 

- How novel is the work? Are the conclusions worth knowing? 

- Is this paper important in its discipline? 

- Are the claims properly placed in the context of the previous literature? 

---

# Points to cover in the review:
### Novelty

- What are the main claims of the paper and how significant are they? 

--

&gt; Our results paint a picture of the combined contributions of migration and selection in shaping the phenotypic landscape of present-day Europeans that suggests __a combination of ancient selection and migration__, rather than recent local selection, is the primary driver of present-day phenotypic differences in Europe.

--

- How novel is the work? Are the conclusions worth knowing? 

&gt; __Ancient DNA (aDNA)__ provides the potential to resolve these issues, by directly observing changes in trait associated allele frequencies over time.

--

&gt; We used __a novel chromosome painting technique__, ... This allows us to identify many trait-associated genetic variants with hitherto unknown evidence for positive selection, as well as __resolve long-standing questions about the timing of selection__ for key health, dietary and pigmentation associated loci.


---

# Points to cover in the review:

### Novelty
- Is this paper important in its discipline? 

&gt; Prehistoric humans underwent three major migration events across Eurasia, ... Each of these events represents a crucial period in human history that has shaped the genetic diversity observed in present-day Europe, ...
- By [Attwaters 2024](https://www.nature.com/articles/s41576-024-00702-4?utm_source=nrg_etoc&amp;utm_medium=email&amp;utm_campaign=toc_41576_25_3&amp;utm_content=20240220) 

--

- Are the claims properly placed in the context of the previous literature? 

&gt; Whilst numerous prior studies have used ancient DNA to infer patterns of selection in Eurasia during the Holocene (e.g., 8–10), many key questions remain unanswered. 

---

# Points to cover in the review:

### Quality

- Do the data and analyses support the authors’ claims?   
  - 1664 _imputed diploid?_ ancient genomes from different times and spaces
  &gt; using imputed aDNA genotype probabilities, we identified 11 genome-wide significant selective sweeps in the GWAS variants

- Is the stated purpose achieved throughout the paper? 

- Would additional work improve the manuscript? 

- Is the experimentation design appropriate for the purpose of the study? 

---

# Results: Figure 1

&lt;div align="center"&gt;
&lt;img src="fig1.png" height=500&gt;
&lt;/div&gt;

---

# Results: Figure 2

&lt;div align="center"&gt;
&lt;img src="fig2.png" height=500&gt;
&lt;/div&gt;

---

# Results: Figure 3

&lt;div align="center"&gt;
&lt;img src="fig3.png" height=500&gt;
&lt;/div&gt;

---

# Results: Figure 4

&lt;div align="center"&gt;
&lt;img src="fig4.png" height=500&gt;
&lt;/div&gt;



---

# Points to cover in the review:

### Clarity

- Evaluate clarity, style and readability of the paper to scientists in the field.  

  - May need an overview diagram to show the sample collection sites.
  - Some fig legends are hard to read

--

- Would you recommend the author seek the service of a professional science writer?

---

# Points to cover in the review:

### Reproducibility

- Are original data (and/or code) deposited in appropriate repositories?

&gt; __Data availability__
All collapsed and paired-end sequence data for novel samples sequenced in this study will be made publicly available on the European Nucleotide Archive, together with trimmed sequence alignment map files, aligned using human build GRCh37. Previously published ancient genomic data used in this study are detailed in Supplementary Table VII of 11, and are all already publicly available.


- Are details of the methodology sufficient to allow the experiments to be reproduced? 

&gt; __Code availability__
The modified version of CLUES used in this study is available from https://github.com/standard-aaron/clues 
...

&gt; All other analyses relied upon available software which has been fully referenced in the manuscript and detailed in the relevant supplementary notes.

    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
